{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "from pathlib import Path"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Mapping"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "DATA_DIR = Path(\"../data\")\n",
    "\n",
    "RUN_DIRS = [run_dir \n",
    " for run_dir in DATA_DIR.glob(\"*\") \n",
    " if run_dir.is_dir() \n",
    " and len(list(run_dir.glob(\"*\"))) != 0 ]\n",
    "\n",
    "MAPPER_WUHAN_DIR = Path(\"../mapper_wuhan\")\n",
    "DEPLETION_DIR = Path(\"../human_depleted_fastqs/depletion_analysis\")\n",
    "GENOME_SARS_PATH = \"../resources/genomes/MN908947_3.fa\"\n",
    "\n",
    "MAPPER_HUMAN_DIR = Path(\"../mapper_human\")\n",
    "GENOME_HUMAN_PATH = \"../resources/genomes/hg38.fa\"\n",
    "\n",
    "MAPPER_HUMAN_SARS_DIR = Path(\"../mapper_human_sarscov2\")\n",
    "GENOME_HUMAN_SARS_PATH = \"../resources/genomes/human_sars.fa\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Mapping vs SARS-CoV Wuhan-Hu-1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "MAPPING against Wuhan-Hu-1 the following runs:\n",
      "- illumina_resv_v2_ns in ../mapper/illumina_resv_v2_ns\n",
      "Sequana command: sequana_mapper --reference /pasteur/projets/policy01/Biomics/resources/covid19/genomes/MN908947_3.fa --input-directory ../data/illumina_resv_v2_ns --do-coverage --working-directory ../mapper/illumina_resv_v2_ns --mapper bowtie2\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mWelcome to Sequana pipelines suite (sequana.readthedocs.io)\u001b[0m\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mWelcome to Sequana pipelines suite (sequana.readthedocs.io)\u001b[0m\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mFound 38 files matching your input  pattern (*fastq.gz)\u001b[0m\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mFound 38 files matching your input  pattern (*fastq.gz)\u001b[0m\n",
      "\u001b[95mCheck the script in ../mapper/illumina_resv_v2_ns/mapper.sh as well as the configuration file in ../mapper/illumina_resv_v2_ns/config.yaml.\n",
      "\u001b[0m\n",
      "\u001b[95mOnce ready, execute the script mapper.sh using \n",
      "\n",
      "\tcd ../mapper/illumina_resv_v2_ns; sbatch mapper.sh\n",
      "\n",
      "\u001b[0m\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mA completion if possible with sequana_completion --name mapper\u001b[0m\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mA completion if possible with sequana_completion --name mapper\u001b[0m\n",
      "\u001b[0mSubmitted batch job 41917320\n",
      "- arbo_single_is in ../mapper/arbo_single_is\n",
      "Sequana command: sequana_mapper --reference /pasteur/projets/policy01/Biomics/resources/covid19/genomes/MN908947_3.fa --input-directory ../data/arbo_single_is --do-coverage --working-directory ../mapper/arbo_single_is --mapper bowtie2\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mWelcome to Sequana pipelines suite (sequana.readthedocs.io)\u001b[0m\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mWelcome to Sequana pipelines suite (sequana.readthedocs.io)\u001b[0m\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mFound 38 files matching your input  pattern (*fastq.gz)\u001b[0m\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mFound 38 files matching your input  pattern (*fastq.gz)\u001b[0m\n",
      "\u001b[95mCheck the script in ../mapper/arbo_single_is/mapper.sh as well as the configuration file in ../mapper/arbo_single_is/config.yaml.\n",
      "\u001b[0m\n",
      "\u001b[95mOnce ready, execute the script mapper.sh using \n",
      "\n",
      "\tcd ../mapper/arbo_single_is; sbatch mapper.sh\n",
      "\n",
      "\u001b[0m\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mA completion if possible with sequana_completion --name mapper\u001b[0m\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mA completion if possible with sequana_completion --name mapper\u001b[0m\n",
      "\u001b[0mSubmitted batch job 41917958\n",
      "- twist_cov_ns in ../mapper/twist_cov_ns\n",
      "Sequana command: sequana_mapper --reference /pasteur/projets/policy01/Biomics/resources/covid19/genomes/MN908947_3.fa --input-directory ../data/twist_cov_ns --do-coverage --working-directory ../mapper/twist_cov_ns --mapper bowtie2\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mWelcome to Sequana pipelines suite (sequana.readthedocs.io)\u001b[0m\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mWelcome to Sequana pipelines suite (sequana.readthedocs.io)\u001b[0m\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mFound 40 files matching your input  pattern (*fastq.gz)\u001b[0m\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mFound 40 files matching your input  pattern (*fastq.gz)\u001b[0m\n",
      "\u001b[95mCheck the script in ../mapper/twist_cov_ns/mapper.sh as well as the configuration file in ../mapper/twist_cov_ns/config.yaml.\n",
      "\u001b[0m\n",
      "\u001b[95mOnce ready, execute the script mapper.sh using \n",
      "\n",
      "\tcd ../mapper/twist_cov_ns; sbatch mapper.sh\n",
      "\n",
      "\u001b[0m\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mA completion if possible with sequana_completion --name mapper\u001b[0m\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mA completion if possible with sequana_completion --name mapper\u001b[0m\n",
      "\u001b[0mSubmitted batch job 41918590\n",
      "- arbo_single_ms in ../mapper/arbo_single_ms\n",
      "Sequana command: sequana_mapper --reference /pasteur/projets/policy01/Biomics/resources/covid19/genomes/MN908947_3.fa --input-directory ../data/arbo_single_ms --do-coverage --working-directory ../mapper/arbo_single_ms --mapper bowtie2\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mWelcome to Sequana pipelines suite (sequana.readthedocs.io)\u001b[0m\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mWelcome to Sequana pipelines suite (sequana.readthedocs.io)\u001b[0m\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mFound 38 files matching your input  pattern (*fastq.gz)\u001b[0m\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mFound 38 files matching your input  pattern (*fastq.gz)\u001b[0m\n",
      "\u001b[95mCheck the script in ../mapper/arbo_single_ms/mapper.sh as well as the configuration file in ../mapper/arbo_single_ms/config.yaml.\n",
      "\u001b[0m\n",
      "\u001b[95mOnce ready, execute the script mapper.sh using \n",
      "\n",
      "\tcd ../mapper/arbo_single_ms; sbatch mapper.sh\n",
      "\n",
      "\u001b[0m\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mA completion if possible with sequana_completion --name mapper\u001b[0m\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mA completion if possible with sequana_completion --name mapper\u001b[0m\n",
      "\u001b[0mSubmitted batch job 41919435\n",
      "- meta_is in ../mapper/meta_is\n",
      "Sequana command: sequana_mapper --reference /pasteur/projets/policy01/Biomics/resources/covid19/genomes/MN908947_3.fa --input-directory ../data/meta_is --do-coverage --working-directory ../mapper/meta_is --mapper bowtie2\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mWelcome to Sequana pipelines suite (sequana.readthedocs.io)\u001b[0m\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mWelcome to Sequana pipelines suite (sequana.readthedocs.io)\u001b[0m\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mFound 42 files matching your input  pattern (*fastq.gz)\u001b[0m\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mFound 42 files matching your input  pattern (*fastq.gz)\u001b[0m\n",
      "\u001b[95mCheck the script in ../mapper/meta_is/mapper.sh as well as the configuration file in ../mapper/meta_is/config.yaml.\n",
      "\u001b[0m\n",
      "\u001b[95mOnce ready, execute the script mapper.sh using \n",
      "\n",
      "\tcd ../mapper/meta_is; sbatch mapper.sh\n",
      "\n",
      "\u001b[0m\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mA completion if possible with sequana_completion --name mapper\u001b[0m\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mA completion if possible with sequana_completion --name mapper\u001b[0m\n",
      "\u001b[0mSubmitted batch job 41920102\n",
      "- arbo_double_10plex_ms in ../mapper/arbo_double_10plex_ms\n",
      "The result folder ../mapper/arbo_double_10plex_ms exists already. Skipping...\n",
      "- twist_cov_hg38_bwt2_depletion_ns in ../mapper/twist_cov_hg38_bwt2_depletion_ns\n",
      "Sequana command: sequana_mapper --reference /pasteur/projets/policy01/Biomics/resources/covid19/genomes/MN908947_3.fa --input-directory ../data/twist_cov_hg38_bwt2_depletion_ns --do-coverage --working-directory ../mapper/twist_cov_hg38_bwt2_depletion_ns --mapper bowtie2\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mWelcome to Sequana pipelines suite (sequana.readthedocs.io)\u001b[0m\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mWelcome to Sequana pipelines suite (sequana.readthedocs.io)\u001b[0m\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mFound 40 files matching your input  pattern (*fastq.gz)\u001b[0m\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mFound 40 files matching your input  pattern (*fastq.gz)\u001b[0m\n",
      "\u001b[95mCheck the script in ../mapper/twist_cov_hg38_bwt2_depletion_ns/mapper.sh as well as the configuration file in ../mapper/twist_cov_hg38_bwt2_depletion_ns/config.yaml.\n",
      "\u001b[0m\n",
      "\u001b[95mOnce ready, execute the script mapper.sh using \n",
      "\n",
      "\tcd ../mapper/twist_cov_hg38_bwt2_depletion_ns; sbatch mapper.sh\n",
      "\n",
      "\u001b[0m\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mA completion if possible with sequana_completion --name mapper\u001b[0m\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mA completion if possible with sequana_completion --name mapper\u001b[0m\n",
      "\u001b[0mSubmitted batch job 41920545\n",
      "- twist_cov_LB_ms in ../mapper/twist_cov_LB_ms\n",
      "The result folder ../mapper/twist_cov_LB_ms exists already. Skipping...\n",
      "- twist_cov_ms in ../mapper/twist_cov_ms\n",
      "Sequana command: sequana_mapper --reference /pasteur/projets/policy01/Biomics/resources/covid19/genomes/MN908947_3.fa --input-directory ../data/twist_cov_ms --do-coverage --working-directory ../mapper/twist_cov_ms --mapper bowtie2\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mWelcome to Sequana pipelines suite (sequana.readthedocs.io)\u001b[0m\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mWelcome to Sequana pipelines suite (sequana.readthedocs.io)\u001b[0m\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mFound 40 files matching your input  pattern (*fastq.gz)\u001b[0m\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mFound 40 files matching your input  pattern (*fastq.gz)\u001b[0m\n",
      "\u001b[95mCheck the script in ../mapper/twist_cov_ms/mapper.sh as well as the configuration file in ../mapper/twist_cov_ms/config.yaml.\n",
      "\u001b[0m\n",
      "\u001b[95mOnce ready, execute the script mapper.sh using \n",
      "\n",
      "\tcd ../mapper/twist_cov_ms; sbatch mapper.sh\n",
      "\n",
      "\u001b[0m\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mA completion if possible with sequana_completion --name mapper\u001b[0m\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mA completion if possible with sequana_completion --name mapper\u001b[0m\n",
      "\u001b[0mSubmitted batch job 41920588\n",
      "- illumina_resv_is in ../mapper/illumina_resv_is\n",
      "Sequana command: sequana_mapper --reference /pasteur/projets/policy01/Biomics/resources/covid19/genomes/MN908947_3.fa --input-directory ../data/illumina_resv_is --do-coverage --working-directory ../mapper/illumina_resv_is --mapper bowtie2\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mWelcome to Sequana pipelines suite (sequana.readthedocs.io)\u001b[0m\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mWelcome to Sequana pipelines suite (sequana.readthedocs.io)\u001b[0m\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mFound 38 files matching your input  pattern (*fastq.gz)\u001b[0m\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mFound 38 files matching your input  pattern (*fastq.gz)\u001b[0m\n",
      "\u001b[95mCheck the script in ../mapper/illumina_resv_is/mapper.sh as well as the configuration file in ../mapper/illumina_resv_is/config.yaml.\n",
      "\u001b[0m\n",
      "\u001b[95mOnce ready, execute the script mapper.sh using \n",
      "\n",
      "\tcd ../mapper/illumina_resv_is; sbatch mapper.sh\n",
      "\n",
      "\u001b[0m\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mA completion if possible with sequana_completion --name mapper\u001b[0m\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mA completion if possible with sequana_completion --name mapper\u001b[0m\n",
      "\u001b[0mSubmitted batch job 41920617\n",
      "- twist_panviral_ns in ../mapper/twist_panviral_ns\n",
      "Sequana command: sequana_mapper --reference /pasteur/projets/policy01/Biomics/resources/covid19/genomes/MN908947_3.fa --input-directory ../data/twist_panviral_ns --do-coverage --working-directory ../mapper/twist_panviral_ns --mapper bowtie2\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mWelcome to Sequana pipelines suite (sequana.readthedocs.io)\u001b[0m\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mWelcome to Sequana pipelines suite (sequana.readthedocs.io)\u001b[0m\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mFound 38 files matching your input  pattern (*fastq.gz)\u001b[0m\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mFound 38 files matching your input  pattern (*fastq.gz)\u001b[0m\n",
      "\u001b[95mCheck the script in ../mapper/twist_panviral_ns/mapper.sh as well as the configuration file in ../mapper/twist_panviral_ns/config.yaml.\n",
      "\u001b[0m\n",
      "\u001b[95mOnce ready, execute the script mapper.sh using \n",
      "\n",
      "\tcd ../mapper/twist_panviral_ns; sbatch mapper.sh\n",
      "\n",
      "\u001b[0m\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mA completion if possible with sequana_completion --name mapper\u001b[0m\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mA completion if possible with sequana_completion --name mapper\u001b[0m\n",
      "\u001b[0mSubmitted batch job 41920639\n",
      "- illumina_resv_ns in ../mapper/illumina_resv_ns\n",
      "Sequana command: sequana_mapper --reference /pasteur/projets/policy01/Biomics/resources/covid19/genomes/MN908947_3.fa --input-directory ../data/illumina_resv_ns --do-coverage --working-directory ../mapper/illumina_resv_ns --mapper bowtie2\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mWelcome to Sequana pipelines suite (sequana.readthedocs.io)\u001b[0m\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mWelcome to Sequana pipelines suite (sequana.readthedocs.io)\u001b[0m\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mFound 38 files matching your input  pattern (*fastq.gz)\u001b[0m\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mFound 38 files matching your input  pattern (*fastq.gz)\u001b[0m\n",
      "\u001b[95mCheck the script in ../mapper/illumina_resv_ns/mapper.sh as well as the configuration file in ../mapper/illumina_resv_ns/config.yaml.\n",
      "\u001b[0m\n",
      "\u001b[95mOnce ready, execute the script mapper.sh using \n",
      "\n",
      "\tcd ../mapper/illumina_resv_ns; sbatch mapper.sh\n",
      "\n",
      "\u001b[0m\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mA completion if possible with sequana_completion --name mapper\u001b[0m\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mA completion if possible with sequana_completion --name mapper\u001b[0m\n",
      "\u001b[0mSubmitted batch job 41920673\n",
      "- arbo_double_ms in ../mapper/arbo_double_ms\n",
      "Sequana command: sequana_mapper --reference /pasteur/projets/policy01/Biomics/resources/covid19/genomes/MN908947_3.fa --input-directory ../data/arbo_double_ms --do-coverage --working-directory ../mapper/arbo_double_ms --mapper bowtie2\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mWelcome to Sequana pipelines suite (sequana.readthedocs.io)\u001b[0m\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mWelcome to Sequana pipelines suite (sequana.readthedocs.io)\u001b[0m\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mFound 38 files matching your input  pattern (*fastq.gz)\u001b[0m\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mFound 38 files matching your input  pattern (*fastq.gz)\u001b[0m\n",
      "\u001b[95mCheck the script in ../mapper/arbo_double_ms/mapper.sh as well as the configuration file in ../mapper/arbo_double_ms/config.yaml.\n",
      "\u001b[0m\n",
      "\u001b[95mOnce ready, execute the script mapper.sh using \n",
      "\n",
      "\tcd ../mapper/arbo_double_ms; sbatch mapper.sh\n",
      "\n",
      "\u001b[0m\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mA completion if possible with sequana_completion --name mapper\u001b[0m\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mA completion if possible with sequana_completion --name mapper\u001b[0m\n",
      "\u001b[0mSubmitted batch job 41920713\n",
      "- illumina_resv_ms in ../mapper/illumina_resv_ms\n",
      "Sequana command: sequana_mapper --reference /pasteur/projets/policy01/Biomics/resources/covid19/genomes/MN908947_3.fa --input-directory ../data/illumina_resv_ms --do-coverage --working-directory ../mapper/illumina_resv_ms --mapper bowtie2\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mWelcome to Sequana pipelines suite (sequana.readthedocs.io)\u001b[0m\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mWelcome to Sequana pipelines suite (sequana.readthedocs.io)\u001b[0m\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mFound 38 files matching your input  pattern (*fastq.gz)\u001b[0m\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mFound 38 files matching your input  pattern (*fastq.gz)\u001b[0m\n",
      "\u001b[95mCheck the script in ../mapper/illumina_resv_ms/mapper.sh as well as the configuration file in ../mapper/illumina_resv_ms/config.yaml.\n",
      "\u001b[0m\n",
      "\u001b[95mOnce ready, execute the script mapper.sh using \n",
      "\n",
      "\tcd ../mapper/illumina_resv_ms; sbatch mapper.sh\n",
      "\n",
      "\u001b[0m\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mA completion if possible with sequana_completion --name mapper\u001b[0m\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mA completion if possible with sequana_completion --name mapper\u001b[0m\n",
      "\u001b[0mSubmitted batch job 41920776\n",
      "- twist_cov_is in ../mapper/twist_cov_is\n",
      "Sequana command: sequana_mapper --reference /pasteur/projets/policy01/Biomics/resources/covid19/genomes/MN908947_3.fa --input-directory ../data/twist_cov_is --do-coverage --working-directory ../mapper/twist_cov_is --mapper bowtie2\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mWelcome to Sequana pipelines suite (sequana.readthedocs.io)\u001b[0m\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mWelcome to Sequana pipelines suite (sequana.readthedocs.io)\u001b[0m\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mFound 40 files matching your input  pattern (*fastq.gz)\u001b[0m\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mFound 40 files matching your input  pattern (*fastq.gz)\u001b[0m\n",
      "\u001b[95mCheck the script in ../mapper/twist_cov_is/mapper.sh as well as the configuration file in ../mapper/twist_cov_is/config.yaml.\n",
      "\u001b[0m\n",
      "\u001b[95mOnce ready, execute the script mapper.sh using \n",
      "\n",
      "\tcd ../mapper/twist_cov_is; sbatch mapper.sh\n",
      "\n",
      "\u001b[0m\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mA completion if possible with sequana_completion --name mapper\u001b[0m\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mA completion if possible with sequana_completion --name mapper\u001b[0m\n",
      "\u001b[0mSubmitted batch job 41920838\n",
      "- twist_cov_10plex_ms in ../mapper/twist_cov_10plex_ms\n",
      "Sequana command: sequana_mapper --reference /pasteur/projets/policy01/Biomics/resources/covid19/genomes/MN908947_3.fa --input-directory ../data/twist_cov_10plex_ms --do-coverage --working-directory ../mapper/twist_cov_10plex_ms --mapper bowtie2\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mWelcome to Sequana pipelines suite (sequana.readthedocs.io)\u001b[0m\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mWelcome to Sequana pipelines suite (sequana.readthedocs.io)\u001b[0m\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mFound 20 files matching your input  pattern (*fastq.gz)\u001b[0m\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mFound 20 files matching your input  pattern (*fastq.gz)\u001b[0m\n",
      "\u001b[95mCheck the script in ../mapper/twist_cov_10plex_ms/mapper.sh as well as the configuration file in ../mapper/twist_cov_10plex_ms/config.yaml.\n",
      "\u001b[0m\n",
      "\u001b[95mOnce ready, execute the script mapper.sh using \n",
      "\n",
      "\tcd ../mapper/twist_cov_10plex_ms; sbatch mapper.sh\n",
      "\n",
      "\u001b[0m\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mA completion if possible with sequana_completion --name mapper\u001b[0m\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mA completion if possible with sequana_completion --name mapper\u001b[0m\n",
      "\u001b[0mSubmitted batch job 41920894\n",
      "- arbo_double_is in ../mapper/arbo_double_is\n",
      "Sequana command: sequana_mapper --reference /pasteur/projets/policy01/Biomics/resources/covid19/genomes/MN908947_3.fa --input-directory ../data/arbo_double_is --do-coverage --working-directory ../mapper/arbo_double_is --mapper bowtie2\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mWelcome to Sequana pipelines suite (sequana.readthedocs.io)\u001b[0m\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mWelcome to Sequana pipelines suite (sequana.readthedocs.io)\u001b[0m\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mFound 38 files matching your input  pattern (*fastq.gz)\u001b[0m\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mFound 38 files matching your input  pattern (*fastq.gz)\u001b[0m\n",
      "\u001b[95mCheck the script in ../mapper/arbo_double_is/mapper.sh as well as the configuration file in ../mapper/arbo_double_is/config.yaml.\n",
      "\u001b[0m\n",
      "\u001b[95mOnce ready, execute the script mapper.sh using \n",
      "\n",
      "\tcd ../mapper/arbo_double_is; sbatch mapper.sh\n",
      "\n",
      "\u001b[0m\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mA completion if possible with sequana_completion --name mapper\u001b[0m\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mA completion if possible with sequana_completion --name mapper\u001b[0m\n",
      "\u001b[0mSubmitted batch job 41920974\n",
      "- meta_ns in ../mapper/meta_ns\n",
      "Sequana command: sequana_mapper --reference /pasteur/projets/policy01/Biomics/resources/covid19/genomes/MN908947_3.fa --input-directory ../data/meta_ns --do-coverage --working-directory ../mapper/meta_ns --mapper bowtie2\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mWelcome to Sequana pipelines suite (sequana.readthedocs.io)\u001b[0m\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mWelcome to Sequana pipelines suite (sequana.readthedocs.io)\u001b[0m\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mFound 42 files matching your input  pattern (*fastq.gz)\u001b[0m\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mFound 42 files matching your input  pattern (*fastq.gz)\u001b[0m\n",
      "\u001b[95mCheck the script in ../mapper/meta_ns/mapper.sh as well as the configuration file in ../mapper/meta_ns/config.yaml.\n",
      "\u001b[0m\n",
      "\u001b[95mOnce ready, execute the script mapper.sh using \n",
      "\n",
      "\tcd ../mapper/meta_ns; sbatch mapper.sh\n",
      "\n",
      "\u001b[0m\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mA completion if possible with sequana_completion --name mapper\u001b[0m\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mA completion if possible with sequana_completion --name mapper\u001b[0m\n",
      "\u001b[0mSubmitted batch job 41921003\n",
      "- twist_panviral_is in ../mapper/twist_panviral_is\n",
      "Sequana command: sequana_mapper --reference /pasteur/projets/policy01/Biomics/resources/covid19/genomes/MN908947_3.fa --input-directory ../data/twist_panviral_is --do-coverage --working-directory ../mapper/twist_panviral_is --mapper bowtie2\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mWelcome to Sequana pipelines suite (sequana.readthedocs.io)\u001b[0m\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mWelcome to Sequana pipelines suite (sequana.readthedocs.io)\u001b[0m\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mFound 38 files matching your input  pattern (*fastq.gz)\u001b[0m\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mFound 38 files matching your input  pattern (*fastq.gz)\u001b[0m\n",
      "\u001b[95mCheck the script in ../mapper/twist_panviral_is/mapper.sh as well as the configuration file in ../mapper/twist_panviral_is/config.yaml.\n",
      "\u001b[0m\n",
      "\u001b[95mOnce ready, execute the script mapper.sh using \n",
      "\n",
      "\tcd ../mapper/twist_panviral_is; sbatch mapper.sh\n",
      "\n",
      "\u001b[0m\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mA completion if possible with sequana_completion --name mapper\u001b[0m\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mA completion if possible with sequana_completion --name mapper\u001b[0m\n",
      "\u001b[0mSubmitted batch job 41921077\n"
     ]
    }
   ],
   "source": [
    "print(\"MAPPING against Wuhan-Hu-1 the following runs:\")\n",
    "for run_dir in RUN_DIRS:\n",
    "    \n",
    "    # Prepare mapper outdir\n",
    "    out_dir = MAPPER_WUHAN_DIR / run_dir.stem\n",
    "    print(f\"- {run_dir.stem} in {out_dir}\")\n",
    "    \n",
    "    # Check if mapping already done:\n",
    "    if out_dir.exists():\n",
    "        print(f\"The result folder {out_dir} exists already. Skipping...\")\n",
    "        continue\n",
    "        \n",
    "    # Prepare sequana command\n",
    "    sequana_cmd = f\"sequana_mapper --reference {GENOME_SARS_PATH} --input-directory {run_dir} --do-coverage --working-directory {out_dir} --mapper bowtie2\"\n",
    "    print(f\"Sequana command: {sequana_cmd}\")\n",
    "    \n",
    "    # Create sequana templates\n",
    "    ! module load sequana/prod && {sequana_cmd}\n",
    "    ! module load sequana/prod && cd {out_dir} && sbatch -q biomics -p biomics -A biomics mapper.sh"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Mapping vs human"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "- illumina_resv_v2_ns in ../mapper_human_unmapped/illumina_resv_v2_ns\n",
      "Sequana command: sequana_mapper --slurm-memory 30G --reference /pasteur/projets/policy01/Biomics/resources/genomes/hg38/hg38.fa --input-directory ../data/illumina_resv_v2_ns --working-directory ../mapper_human_unmapped/illumina_resv_v2_ns --mapper bowtie2\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mWelcome to Sequana pipelines suite (sequana.readthedocs.io)\u001b[0m\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mWelcome to Sequana pipelines suite (sequana.readthedocs.io)\u001b[0m\n",
      "Traceback (most recent call last):\n",
      "  File \"/pasteur/projets/specific/Biomics/resources/condas/sequana_0_9_5/bin/sequana_mapper\", line 8, in <module>\n",
      "    sys.exit(main())\n",
      "  File \"/pasteur/projets/policy01/Biomics/resources/condas/sequana_0_9_5/lib/python3.7/site-packages/sequana_pipelines/mapper/main.py\", line 82, in main\n",
      "    manager.setup()\n",
      "  File \"/pasteur/projets/policy01/Biomics/resources/condas/sequana_0_9_5/lib/python3.7/site-packages/sequana-0.9.5-py3.7.egg/sequana/pipelines_common.py\", line 700, in setup\n",
      "    self._create_directories()\n",
      "  File \"/pasteur/projets/policy01/Biomics/resources/condas/sequana_0_9_5/lib/python3.7/site-packages/sequana-0.9.5-py3.7.egg/sequana/pipelines_common.py\", line 713, in _create_directories\n",
      "    os.mkdir(self.workdir)\n",
      "FileNotFoundError: [Errno 2] No such file or directory: '../mapper_human_unmapped/illumina_resv_v2_ns'\n",
      "\u001b[0m- arbo_single_is in ../mapper_human_unmapped/arbo_single_is\n",
      "Sequana command: sequana_mapper --slurm-memory 30G --reference /pasteur/projets/policy01/Biomics/resources/genomes/hg38/hg38.fa --input-directory ../data/arbo_single_is --working-directory ../mapper_human_unmapped/arbo_single_is --mapper bowtie2\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mWelcome to Sequana pipelines suite (sequana.readthedocs.io)\u001b[0m\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mWelcome to Sequana pipelines suite (sequana.readthedocs.io)\u001b[0m\n",
      "Traceback (most recent call last):\n",
      "  File \"/pasteur/projets/specific/Biomics/resources/condas/sequana_0_9_5/bin/sequana_mapper\", line 8, in <module>\n",
      "    sys.exit(main())\n",
      "  File \"/pasteur/projets/policy01/Biomics/resources/condas/sequana_0_9_5/lib/python3.7/site-packages/sequana_pipelines/mapper/main.py\", line 82, in main\n",
      "    manager.setup()\n",
      "  File \"/pasteur/projets/policy01/Biomics/resources/condas/sequana_0_9_5/lib/python3.7/site-packages/sequana-0.9.5-py3.7.egg/sequana/pipelines_common.py\", line 700, in setup\n",
      "    self._create_directories()\n",
      "  File \"/pasteur/projets/policy01/Biomics/resources/condas/sequana_0_9_5/lib/python3.7/site-packages/sequana-0.9.5-py3.7.egg/sequana/pipelines_common.py\", line 713, in _create_directories\n",
      "    os.mkdir(self.workdir)\n",
      "FileNotFoundError: [Errno 2] No such file or directory: '../mapper_human_unmapped/arbo_single_is'\n",
      "\u001b[0m- twist_cov_ns in ../mapper_human_unmapped/twist_cov_ns\n",
      "Sequana command: sequana_mapper --slurm-memory 30G --reference /pasteur/projets/policy01/Biomics/resources/genomes/hg38/hg38.fa --input-directory ../data/twist_cov_ns --working-directory ../mapper_human_unmapped/twist_cov_ns --mapper bowtie2\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mWelcome to Sequana pipelines suite (sequana.readthedocs.io)\u001b[0m\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mWelcome to Sequana pipelines suite (sequana.readthedocs.io)\u001b[0m\n",
      "Traceback (most recent call last):\n",
      "  File \"/pasteur/projets/specific/Biomics/resources/condas/sequana_0_9_5/bin/sequana_mapper\", line 8, in <module>\n",
      "    sys.exit(main())\n",
      "  File \"/pasteur/projets/policy01/Biomics/resources/condas/sequana_0_9_5/lib/python3.7/site-packages/sequana_pipelines/mapper/main.py\", line 82, in main\n",
      "    manager.setup()\n",
      "  File \"/pasteur/projets/policy01/Biomics/resources/condas/sequana_0_9_5/lib/python3.7/site-packages/sequana-0.9.5-py3.7.egg/sequana/pipelines_common.py\", line 700, in setup\n",
      "    self._create_directories()\n",
      "  File \"/pasteur/projets/policy01/Biomics/resources/condas/sequana_0_9_5/lib/python3.7/site-packages/sequana-0.9.5-py3.7.egg/sequana/pipelines_common.py\", line 713, in _create_directories\n",
      "    os.mkdir(self.workdir)\n",
      "FileNotFoundError: [Errno 2] No such file or directory: '../mapper_human_unmapped/twist_cov_ns'\n",
      "\u001b[0m- arbo_single_ms in ../mapper_human_unmapped/arbo_single_ms\n",
      "Sequana command: sequana_mapper --slurm-memory 30G --reference /pasteur/projets/policy01/Biomics/resources/genomes/hg38/hg38.fa --input-directory ../data/arbo_single_ms --working-directory ../mapper_human_unmapped/arbo_single_ms --mapper bowtie2\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mWelcome to Sequana pipelines suite (sequana.readthedocs.io)\u001b[0m\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mWelcome to Sequana pipelines suite (sequana.readthedocs.io)\u001b[0m\n",
      "Traceback (most recent call last):\n",
      "  File \"/pasteur/projets/specific/Biomics/resources/condas/sequana_0_9_5/bin/sequana_mapper\", line 8, in <module>\n",
      "    sys.exit(main())\n",
      "  File \"/pasteur/projets/policy01/Biomics/resources/condas/sequana_0_9_5/lib/python3.7/site-packages/sequana_pipelines/mapper/main.py\", line 82, in main\n",
      "    manager.setup()\n",
      "  File \"/pasteur/projets/policy01/Biomics/resources/condas/sequana_0_9_5/lib/python3.7/site-packages/sequana-0.9.5-py3.7.egg/sequana/pipelines_common.py\", line 700, in setup\n",
      "    self._create_directories()\n",
      "  File \"/pasteur/projets/policy01/Biomics/resources/condas/sequana_0_9_5/lib/python3.7/site-packages/sequana-0.9.5-py3.7.egg/sequana/pipelines_common.py\", line 713, in _create_directories\n",
      "    os.mkdir(self.workdir)\n",
      "FileNotFoundError: [Errno 2] No such file or directory: '../mapper_human_unmapped/arbo_single_ms'\n",
      "\u001b[0m- meta_is in ../mapper_human_unmapped/meta_is\n",
      "Sequana command: sequana_mapper --slurm-memory 30G --reference /pasteur/projets/policy01/Biomics/resources/genomes/hg38/hg38.fa --input-directory ../data/meta_is --working-directory ../mapper_human_unmapped/meta_is --mapper bowtie2\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mWelcome to Sequana pipelines suite (sequana.readthedocs.io)\u001b[0m\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mWelcome to Sequana pipelines suite (sequana.readthedocs.io)\u001b[0m\n",
      "Traceback (most recent call last):\n",
      "  File \"/pasteur/projets/specific/Biomics/resources/condas/sequana_0_9_5/bin/sequana_mapper\", line 8, in <module>\n",
      "    sys.exit(main())\n",
      "  File \"/pasteur/projets/policy01/Biomics/resources/condas/sequana_0_9_5/lib/python3.7/site-packages/sequana_pipelines/mapper/main.py\", line 82, in main\n",
      "    manager.setup()\n",
      "  File \"/pasteur/projets/policy01/Biomics/resources/condas/sequana_0_9_5/lib/python3.7/site-packages/sequana-0.9.5-py3.7.egg/sequana/pipelines_common.py\", line 700, in setup\n",
      "    self._create_directories()\n",
      "  File \"/pasteur/projets/policy01/Biomics/resources/condas/sequana_0_9_5/lib/python3.7/site-packages/sequana-0.9.5-py3.7.egg/sequana/pipelines_common.py\", line 713, in _create_directories\n",
      "    os.mkdir(self.workdir)\n",
      "FileNotFoundError: [Errno 2] No such file or directory: '../mapper_human_unmapped/meta_is'\n",
      "\u001b[0m- arbo_double_10plex_ms in ../mapper_human_unmapped/arbo_double_10plex_ms\n",
      "Sequana command: sequana_mapper --slurm-memory 30G --reference /pasteur/projets/policy01/Biomics/resources/genomes/hg38/hg38.fa --input-directory ../data/arbo_double_10plex_ms --working-directory ../mapper_human_unmapped/arbo_double_10plex_ms --mapper bowtie2\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mWelcome to Sequana pipelines suite (sequana.readthedocs.io)\u001b[0m\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mWelcome to Sequana pipelines suite (sequana.readthedocs.io)\u001b[0m\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mFound 20 files matching your input  pattern (*fastq.gz)\u001b[0m\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mFound 20 files matching your input  pattern (*fastq.gz)\u001b[0m\n",
      "\u001b[95mCheck the script in ../mapper_human_unmapped/arbo_double_10plex_ms/mapper.sh as well as the configuration file in ../mapper_human_unmapped/arbo_double_10plex_ms/config.yaml.\n",
      "\u001b[0m\n",
      "\u001b[95mOnce ready, execute the script mapper.sh using \n",
      "\n",
      "\tcd ../mapper_human_unmapped/arbo_double_10plex_ms; sbatch mapper.sh\n",
      "\n",
      "\u001b[0m\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mA completion if possible with sequana_completion --name mapper\u001b[0m\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mA completion if possible with sequana_completion --name mapper\u001b[0m\n",
      "\u001b[0m- twist_cov_hg38_bwt2_depletion_ns in ../mapper_human_unmapped/twist_cov_hg38_bwt2_depletion_ns\n",
      "Sequana command: sequana_mapper --slurm-memory 30G --reference /pasteur/projets/policy01/Biomics/resources/genomes/hg38/hg38.fa --input-directory ../data/twist_cov_hg38_bwt2_depletion_ns --working-directory ../mapper_human_unmapped/twist_cov_hg38_bwt2_depletion_ns --mapper bowtie2\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mWelcome to Sequana pipelines suite (sequana.readthedocs.io)\u001b[0m\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mWelcome to Sequana pipelines suite (sequana.readthedocs.io)\u001b[0m\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mFound 40 files matching your input  pattern (*fastq.gz)\u001b[0m\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mFound 40 files matching your input  pattern (*fastq.gz)\u001b[0m\n",
      "\u001b[95mCheck the script in ../mapper_human_unmapped/twist_cov_hg38_bwt2_depletion_ns/mapper.sh as well as the configuration file in ../mapper_human_unmapped/twist_cov_hg38_bwt2_depletion_ns/config.yaml.\n",
      "\u001b[0m\n",
      "\u001b[95mOnce ready, execute the script mapper.sh using \n",
      "\n",
      "\tcd ../mapper_human_unmapped/twist_cov_hg38_bwt2_depletion_ns; sbatch mapper.sh\n",
      "\n",
      "\u001b[0m\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mA completion if possible with sequana_completion --name mapper\u001b[0m\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mA completion if possible with sequana_completion --name mapper\u001b[0m\n",
      "\u001b[0m- twist_cov_LB_ms in ../mapper_human_unmapped/twist_cov_LB_ms\n",
      "Sequana command: sequana_mapper --slurm-memory 30G --reference /pasteur/projets/policy01/Biomics/resources/genomes/hg38/hg38.fa --input-directory ../data/twist_cov_LB_ms --working-directory ../mapper_human_unmapped/twist_cov_LB_ms --mapper bowtie2\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mWelcome to Sequana pipelines suite (sequana.readthedocs.io)\u001b[0m\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mWelcome to Sequana pipelines suite (sequana.readthedocs.io)\u001b[0m\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mFound 40 files matching your input  pattern (*fastq.gz)\u001b[0m\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mFound 40 files matching your input  pattern (*fastq.gz)\u001b[0m\n",
      "\u001b[95mCheck the script in ../mapper_human_unmapped/twist_cov_LB_ms/mapper.sh as well as the configuration file in ../mapper_human_unmapped/twist_cov_LB_ms/config.yaml.\n",
      "\u001b[0m\n",
      "\u001b[95mOnce ready, execute the script mapper.sh using \n",
      "\n",
      "\tcd ../mapper_human_unmapped/twist_cov_LB_ms; sbatch mapper.sh\n",
      "\n",
      "\u001b[0m\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mA completion if possible with sequana_completion --name mapper\u001b[0m\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mA completion if possible with sequana_completion --name mapper\u001b[0m\n",
      "\u001b[0m- twist_cov_ms in ../mapper_human_unmapped/twist_cov_ms\n",
      "Sequana command: sequana_mapper --slurm-memory 30G --reference /pasteur/projets/policy01/Biomics/resources/genomes/hg38/hg38.fa --input-directory ../data/twist_cov_ms --working-directory ../mapper_human_unmapped/twist_cov_ms --mapper bowtie2\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mWelcome to Sequana pipelines suite (sequana.readthedocs.io)\u001b[0m\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mWelcome to Sequana pipelines suite (sequana.readthedocs.io)\u001b[0m\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mFound 40 files matching your input  pattern (*fastq.gz)\u001b[0m\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mFound 40 files matching your input  pattern (*fastq.gz)\u001b[0m\n",
      "\u001b[95mCheck the script in ../mapper_human_unmapped/twist_cov_ms/mapper.sh as well as the configuration file in ../mapper_human_unmapped/twist_cov_ms/config.yaml.\n",
      "\u001b[0m\n",
      "\u001b[95mOnce ready, execute the script mapper.sh using \n",
      "\n",
      "\tcd ../mapper_human_unmapped/twist_cov_ms; sbatch mapper.sh\n",
      "\n",
      "\u001b[0m\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mA completion if possible with sequana_completion --name mapper\u001b[0m\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mA completion if possible with sequana_completion --name mapper\u001b[0m\n",
      "\u001b[0m- illumina_resv_is in ../mapper_human_unmapped/illumina_resv_is\n",
      "Sequana command: sequana_mapper --slurm-memory 30G --reference /pasteur/projets/policy01/Biomics/resources/genomes/hg38/hg38.fa --input-directory ../data/illumina_resv_is --working-directory ../mapper_human_unmapped/illumina_resv_is --mapper bowtie2\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mWelcome to Sequana pipelines suite (sequana.readthedocs.io)\u001b[0m\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mWelcome to Sequana pipelines suite (sequana.readthedocs.io)\u001b[0m\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mFound 38 files matching your input  pattern (*fastq.gz)\u001b[0m\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mFound 38 files matching your input  pattern (*fastq.gz)\u001b[0m\n",
      "\u001b[95mCheck the script in ../mapper_human_unmapped/illumina_resv_is/mapper.sh as well as the configuration file in ../mapper_human_unmapped/illumina_resv_is/config.yaml.\n",
      "\u001b[0m\n",
      "\u001b[95mOnce ready, execute the script mapper.sh using \n",
      "\n",
      "\tcd ../mapper_human_unmapped/illumina_resv_is; sbatch mapper.sh\n",
      "\n",
      "\u001b[0m\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mA completion if possible with sequana_completion --name mapper\u001b[0m\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mA completion if possible with sequana_completion --name mapper\u001b[0m\n",
      "\u001b[0m- twist_panviral_ns in ../mapper_human_unmapped/twist_panviral_ns\n",
      "Sequana command: sequana_mapper --slurm-memory 30G --reference /pasteur/projets/policy01/Biomics/resources/genomes/hg38/hg38.fa --input-directory ../data/twist_panviral_ns --working-directory ../mapper_human_unmapped/twist_panviral_ns --mapper bowtie2\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mWelcome to Sequana pipelines suite (sequana.readthedocs.io)\u001b[0m\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mWelcome to Sequana pipelines suite (sequana.readthedocs.io)\u001b[0m\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mFound 38 files matching your input  pattern (*fastq.gz)\u001b[0m\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mFound 38 files matching your input  pattern (*fastq.gz)\u001b[0m\n",
      "\u001b[95mCheck the script in ../mapper_human_unmapped/twist_panviral_ns/mapper.sh as well as the configuration file in ../mapper_human_unmapped/twist_panviral_ns/config.yaml.\n",
      "\u001b[0m\n",
      "\u001b[95mOnce ready, execute the script mapper.sh using \n",
      "\n",
      "\tcd ../mapper_human_unmapped/twist_panviral_ns; sbatch mapper.sh\n",
      "\n",
      "\u001b[0m\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mA completion if possible with sequana_completion --name mapper\u001b[0m\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mA completion if possible with sequana_completion --name mapper\u001b[0m\n",
      "\u001b[0m- illumina_resv_ns in ../mapper_human_unmapped/illumina_resv_ns\n",
      "Sequana command: sequana_mapper --slurm-memory 30G --reference /pasteur/projets/policy01/Biomics/resources/genomes/hg38/hg38.fa --input-directory ../data/illumina_resv_ns --working-directory ../mapper_human_unmapped/illumina_resv_ns --mapper bowtie2\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mWelcome to Sequana pipelines suite (sequana.readthedocs.io)\u001b[0m\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mWelcome to Sequana pipelines suite (sequana.readthedocs.io)\u001b[0m\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mFound 38 files matching your input  pattern (*fastq.gz)\u001b[0m\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mFound 38 files matching your input  pattern (*fastq.gz)\u001b[0m\n",
      "\u001b[95mCheck the script in ../mapper_human_unmapped/illumina_resv_ns/mapper.sh as well as the configuration file in ../mapper_human_unmapped/illumina_resv_ns/config.yaml.\n",
      "\u001b[0m\n",
      "\u001b[95mOnce ready, execute the script mapper.sh using \n",
      "\n",
      "\tcd ../mapper_human_unmapped/illumina_resv_ns; sbatch mapper.sh\n",
      "\n",
      "\u001b[0m\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mA completion if possible with sequana_completion --name mapper\u001b[0m\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mA completion if possible with sequana_completion --name mapper\u001b[0m\n",
      "\u001b[0m- arbo_double_ms in ../mapper_human_unmapped/arbo_double_ms\n",
      "Sequana command: sequana_mapper --slurm-memory 30G --reference /pasteur/projets/policy01/Biomics/resources/genomes/hg38/hg38.fa --input-directory ../data/arbo_double_ms --working-directory ../mapper_human_unmapped/arbo_double_ms --mapper bowtie2\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mWelcome to Sequana pipelines suite (sequana.readthedocs.io)\u001b[0m\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mWelcome to Sequana pipelines suite (sequana.readthedocs.io)\u001b[0m\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mFound 38 files matching your input  pattern (*fastq.gz)\u001b[0m\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mFound 38 files matching your input  pattern (*fastq.gz)\u001b[0m\n",
      "\u001b[95mCheck the script in ../mapper_human_unmapped/arbo_double_ms/mapper.sh as well as the configuration file in ../mapper_human_unmapped/arbo_double_ms/config.yaml.\n",
      "\u001b[0m\n",
      "\u001b[95mOnce ready, execute the script mapper.sh using \n",
      "\n",
      "\tcd ../mapper_human_unmapped/arbo_double_ms; sbatch mapper.sh\n",
      "\n",
      "\u001b[0m\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mA completion if possible with sequana_completion --name mapper\u001b[0m\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mA completion if possible with sequana_completion --name mapper\u001b[0m\n",
      "\u001b[0m- illumina_resv_ms in ../mapper_human_unmapped/illumina_resv_ms\n",
      "Sequana command: sequana_mapper --slurm-memory 30G --reference /pasteur/projets/policy01/Biomics/resources/genomes/hg38/hg38.fa --input-directory ../data/illumina_resv_ms --working-directory ../mapper_human_unmapped/illumina_resv_ms --mapper bowtie2\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mWelcome to Sequana pipelines suite (sequana.readthedocs.io)\u001b[0m\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mWelcome to Sequana pipelines suite (sequana.readthedocs.io)\u001b[0m\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mFound 38 files matching your input  pattern (*fastq.gz)\u001b[0m\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mFound 38 files matching your input  pattern (*fastq.gz)\u001b[0m\n",
      "\u001b[95mCheck the script in ../mapper_human_unmapped/illumina_resv_ms/mapper.sh as well as the configuration file in ../mapper_human_unmapped/illumina_resv_ms/config.yaml.\n",
      "\u001b[0m\n",
      "\u001b[95mOnce ready, execute the script mapper.sh using \n",
      "\n",
      "\tcd ../mapper_human_unmapped/illumina_resv_ms; sbatch mapper.sh\n",
      "\n",
      "\u001b[0m\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mA completion if possible with sequana_completion --name mapper\u001b[0m\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mA completion if possible with sequana_completion --name mapper\u001b[0m\n",
      "\u001b[0m- twist_cov_is in ../mapper_human_unmapped/twist_cov_is\n",
      "Sequana command: sequana_mapper --slurm-memory 30G --reference /pasteur/projets/policy01/Biomics/resources/genomes/hg38/hg38.fa --input-directory ../data/twist_cov_is --working-directory ../mapper_human_unmapped/twist_cov_is --mapper bowtie2\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mWelcome to Sequana pipelines suite (sequana.readthedocs.io)\u001b[0m\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mWelcome to Sequana pipelines suite (sequana.readthedocs.io)\u001b[0m\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mFound 40 files matching your input  pattern (*fastq.gz)\u001b[0m\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mFound 40 files matching your input  pattern (*fastq.gz)\u001b[0m\n",
      "\u001b[95mCheck the script in ../mapper_human_unmapped/twist_cov_is/mapper.sh as well as the configuration file in ../mapper_human_unmapped/twist_cov_is/config.yaml.\n",
      "\u001b[0m\n",
      "\u001b[95mOnce ready, execute the script mapper.sh using \n",
      "\n",
      "\tcd ../mapper_human_unmapped/twist_cov_is; sbatch mapper.sh\n",
      "\n",
      "\u001b[0m\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mA completion if possible with sequana_completion --name mapper\u001b[0m\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mA completion if possible with sequana_completion --name mapper\u001b[0m\n",
      "\u001b[0m- twist_cov_10plex_ms in ../mapper_human_unmapped/twist_cov_10plex_ms\n",
      "Sequana command: sequana_mapper --slurm-memory 30G --reference /pasteur/projets/policy01/Biomics/resources/genomes/hg38/hg38.fa --input-directory ../data/twist_cov_10plex_ms --working-directory ../mapper_human_unmapped/twist_cov_10plex_ms --mapper bowtie2\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mWelcome to Sequana pipelines suite (sequana.readthedocs.io)\u001b[0m\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mWelcome to Sequana pipelines suite (sequana.readthedocs.io)\u001b[0m\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mFound 20 files matching your input  pattern (*fastq.gz)\u001b[0m\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mFound 20 files matching your input  pattern (*fastq.gz)\u001b[0m\n",
      "\u001b[95mCheck the script in ../mapper_human_unmapped/twist_cov_10plex_ms/mapper.sh as well as the configuration file in ../mapper_human_unmapped/twist_cov_10plex_ms/config.yaml.\n",
      "\u001b[0m\n",
      "\u001b[95mOnce ready, execute the script mapper.sh using \n",
      "\n",
      "\tcd ../mapper_human_unmapped/twist_cov_10plex_ms; sbatch mapper.sh\n",
      "\n",
      "\u001b[0m\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mA completion if possible with sequana_completion --name mapper\u001b[0m\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mA completion if possible with sequana_completion --name mapper\u001b[0m\n",
      "\u001b[0m- arbo_double_is in ../mapper_human_unmapped/arbo_double_is\n",
      "Sequana command: sequana_mapper --slurm-memory 30G --reference /pasteur/projets/policy01/Biomics/resources/genomes/hg38/hg38.fa --input-directory ../data/arbo_double_is --working-directory ../mapper_human_unmapped/arbo_double_is --mapper bowtie2\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mWelcome to Sequana pipelines suite (sequana.readthedocs.io)\u001b[0m\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mWelcome to Sequana pipelines suite (sequana.readthedocs.io)\u001b[0m\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mFound 38 files matching your input  pattern (*fastq.gz)\u001b[0m\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mFound 38 files matching your input  pattern (*fastq.gz)\u001b[0m\n",
      "\u001b[95mCheck the script in ../mapper_human_unmapped/arbo_double_is/mapper.sh as well as the configuration file in ../mapper_human_unmapped/arbo_double_is/config.yaml.\n",
      "\u001b[0m\n",
      "\u001b[95mOnce ready, execute the script mapper.sh using \n",
      "\n",
      "\tcd ../mapper_human_unmapped/arbo_double_is; sbatch mapper.sh\n",
      "\n",
      "\u001b[0m\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mA completion if possible with sequana_completion --name mapper\u001b[0m\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mA completion if possible with sequana_completion --name mapper\u001b[0m\n",
      "\u001b[0m- meta_ns in ../mapper_human_unmapped/meta_ns\n",
      "Sequana command: sequana_mapper --slurm-memory 30G --reference /pasteur/projets/policy01/Biomics/resources/genomes/hg38/hg38.fa --input-directory ../data/meta_ns --working-directory ../mapper_human_unmapped/meta_ns --mapper bowtie2\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mWelcome to Sequana pipelines suite (sequana.readthedocs.io)\u001b[0m\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mWelcome to Sequana pipelines suite (sequana.readthedocs.io)\u001b[0m\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mFound 42 files matching your input  pattern (*fastq.gz)\u001b[0m\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mFound 42 files matching your input  pattern (*fastq.gz)\u001b[0m\n",
      "\u001b[95mCheck the script in ../mapper_human_unmapped/meta_ns/mapper.sh as well as the configuration file in ../mapper_human_unmapped/meta_ns/config.yaml.\n",
      "\u001b[0m\n",
      "\u001b[95mOnce ready, execute the script mapper.sh using \n",
      "\n",
      "\tcd ../mapper_human_unmapped/meta_ns; sbatch mapper.sh\n",
      "\n",
      "\u001b[0m\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mA completion if possible with sequana_completion --name mapper\u001b[0m\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mA completion if possible with sequana_completion --name mapper\u001b[0m\n",
      "\u001b[0m- twist_panviral_is in ../mapper_human_unmapped/twist_panviral_is\n",
      "Sequana command: sequana_mapper --slurm-memory 30G --reference /pasteur/projets/policy01/Biomics/resources/genomes/hg38/hg38.fa --input-directory ../data/twist_panviral_is --working-directory ../mapper_human_unmapped/twist_panviral_is --mapper bowtie2\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mWelcome to Sequana pipelines suite (sequana.readthedocs.io)\u001b[0m\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mWelcome to Sequana pipelines suite (sequana.readthedocs.io)\u001b[0m\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mFound 38 files matching your input  pattern (*fastq.gz)\u001b[0m\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mFound 38 files matching your input  pattern (*fastq.gz)\u001b[0m\n",
      "\u001b[95mCheck the script in ../mapper_human_unmapped/twist_panviral_is/mapper.sh as well as the configuration file in ../mapper_human_unmapped/twist_panviral_is/config.yaml.\n",
      "\u001b[0m\n",
      "\u001b[95mOnce ready, execute the script mapper.sh using \n",
      "\n",
      "\tcd ../mapper_human_unmapped/twist_panviral_is; sbatch mapper.sh\n",
      "\n",
      "\u001b[0m\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mA completion if possible with sequana_completion --name mapper\u001b[0m\n",
      "\u001b[32mINFO    [sequana.pipelines_common]: \u001b[0m \u001b[34mA completion if possible with sequana_completion --name mapper\u001b[0m\n",
      "\u001b[0m"
     ]
    }
   ],
   "source": [
    "for run_dir in RUN_DIRS:\n",
    "    \n",
    "    # Prepare mapper outdir\n",
    "    out_dir = MAPPER_HUMAN_DIR / run_dir.stem\n",
    "    print(f\"- {run_dir.stem} in {out_dir}\")\n",
    "    \n",
    "    # Check if mapping already done:\n",
    "    if out_dir.exists():\n",
    "        print(f\"The result folder {out_dir} exists already. Skipping...\")\n",
    "        continue\n",
    "        \n",
    "    # Prepare sequana command\n",
    "    sequana_cmd = f\"sequana_mapper --slurm-memory 30G --reference {GENOME_HUMAN_PATH} --input-directory {run_dir} --working-directory {out_dir} --mapper bowtie2\"\n",
    "    print(f\"Sequana command: {sequana_cmd}\")\n",
    "    \n",
    "    # Create sequana templates\n",
    "    ! module load sequana/prod && {sequana_cmd}\n",
    "    ! module load sequana/prod && cd {out_dir} && sbatch -q biomics -p biomics -A biomics mapper.sh"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## In silico depletion"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "- arbo_single_is in ../human_depleted_fastqs/arbo_single_is\n",
      "../human_depleted_fastqs/arbo_single_is/4660_S3_L001/bwa_mem_mapping/4660_S3_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 59788 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 59788 reads\n",
      "../human_depleted_fastqs/arbo_single_is/4798_S16_L001/bwa_mem_mapping/4798_S16_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 262 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 262 reads\n",
      "../human_depleted_fastqs/arbo_single_is/4797_S17_L001/bwa_mem_mapping/4797_S17_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 314 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 314 reads\n",
      "../human_depleted_fastqs/arbo_single_is/4656_S18_L001/bwa_mem_mapping/4656_S18_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 44246 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 44246 reads\n",
      "../human_depleted_fastqs/arbo_single_is/4777_S13_L001/bwa_mem_mapping/4777_S13_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 7850 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 7850 reads\n",
      "../human_depleted_fastqs/arbo_single_is/4716_S2_L001/bwa_mem_mapping/4716_S2_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 14910 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 14910 reads\n",
      "../human_depleted_fastqs/arbo_single_is/4489_S15_L001/bwa_mem_mapping/4489_S15_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 630 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 630 reads\n",
      "../human_depleted_fastqs/arbo_single_is/4688_S11_L001/bwa_mem_mapping/4688_S11_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 17828 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 17828 reads\n",
      "../human_depleted_fastqs/arbo_single_is/4676_S7_L001/bwa_mem_mapping/4676_S7_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 26868 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 26868 reads\n",
      "../human_depleted_fastqs/arbo_single_is/4653_S8_L001/bwa_mem_mapping/4653_S8_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 66842 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 66842 reads\n",
      "../human_depleted_fastqs/arbo_single_is/4520_S4_L001/bwa_mem_mapping/4520_S4_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 57142 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 57142 reads\n",
      "../human_depleted_fastqs/arbo_single_is/4673_S12_L001/bwa_mem_mapping/4673_S12_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 43104 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 43104 reads\n",
      "../human_depleted_fastqs/arbo_single_is/4697_S6_L001/bwa_mem_mapping/4697_S6_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 19186 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 19186 reads\n",
      "../human_depleted_fastqs/arbo_single_is/4787_S10_L001/bwa_mem_mapping/4787_S10_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 7556 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 7556 reads\n",
      "../human_depleted_fastqs/arbo_single_is/4885_S1_L001/bwa_mem_mapping/4885_S1_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 7210 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 7210 reads\n",
      "../human_depleted_fastqs/arbo_single_is/4707_S5_L001/bwa_mem_mapping/4707_S5_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 5114 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 5114 reads\n",
      "../human_depleted_fastqs/arbo_single_is/4544_S19_L001/bwa_mem_mapping/4544_S19_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 1016 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 1016 reads\n",
      "../human_depleted_fastqs/arbo_single_is/4861_S9_L001/bwa_mem_mapping/4861_S9_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 8068 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 8068 reads\n",
      "../human_depleted_fastqs/arbo_single_is/4668_S14_L001/bwa_mem_mapping/4668_S14_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 18518 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 18518 reads\n",
      "- twist_cov_ns in ../human_depleted_fastqs/twist_cov_ns\n",
      "../human_depleted_fastqs/twist_cov_ns/4489_S15/bwa_mem_mapping/4489_S15.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 622706 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 622706 reads\n",
      "../human_depleted_fastqs/twist_cov_ns/4656_S18/bwa_mem_mapping/4656_S18.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 12857132 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 12857132 reads\n",
      "../human_depleted_fastqs/twist_cov_ns/4797_S17/bwa_mem_mapping/4797_S17.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 16960 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 16960 reads\n",
      "../human_depleted_fastqs/twist_cov_ns/4520_S4/bwa_mem_mapping/4520_S4.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 11747538 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 11747538 reads\n",
      "../human_depleted_fastqs/twist_cov_ns/4787_S10/bwa_mem_mapping/4787_S10.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 371324 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 371324 reads\n",
      "../human_depleted_fastqs/twist_cov_ns/4777_S13/bwa_mem_mapping/4777_S13.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 724616 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 724616 reads\n",
      "../human_depleted_fastqs/twist_cov_ns/4697_S6/bwa_mem_mapping/4697_S6.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 3127852 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 3127852 reads\n",
      "../human_depleted_fastqs/twist_cov_ns/4707_S5/bwa_mem_mapping/4707_S5.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 1200738 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 1200738 reads\n",
      "../human_depleted_fastqs/twist_cov_ns/4688_S11/bwa_mem_mapping/4688_S11.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 4270420 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 4270420 reads\n",
      "../human_depleted_fastqs/twist_cov_ns/4861_S9/bwa_mem_mapping/4861_S9.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 7978708 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 7978708 reads\n",
      "../human_depleted_fastqs/twist_cov_ns/4673_S12/bwa_mem_mapping/4673_S12.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 8314254 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 8314254 reads\n",
      "../human_depleted_fastqs/twist_cov_ns/4885_S1/bwa_mem_mapping/4885_S1.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 1519738 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 1519738 reads\n",
      "../human_depleted_fastqs/twist_cov_ns/4660_S3/bwa_mem_mapping/4660_S3.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 11648232 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 11648232 reads\n",
      "../human_depleted_fastqs/twist_cov_ns/4544_S19/bwa_mem_mapping/4544_S19.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 527720 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 527720 reads\n",
      "../human_depleted_fastqs/twist_cov_ns/4510_S20/bwa_mem_mapping/4510_S20.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 6082408 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 6082408 reads\n",
      "../human_depleted_fastqs/twist_cov_ns/4798_S16/bwa_mem_mapping/4798_S16.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 15998 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 15998 reads\n",
      "../human_depleted_fastqs/twist_cov_ns/4668_S14/bwa_mem_mapping/4668_S14.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 1102558 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 1102558 reads\n",
      "../human_depleted_fastqs/twist_cov_ns/4716_S2/bwa_mem_mapping/4716_S2.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 2255376 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 2255376 reads\n",
      "../human_depleted_fastqs/twist_cov_ns/4676_S7/bwa_mem_mapping/4676_S7.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 6054474 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 6054474 reads\n",
      "../human_depleted_fastqs/twist_cov_ns/4653_S8/bwa_mem_mapping/4653_S8.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 19781506 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 19781506 reads\n",
      "- arbo_single_ms in ../human_depleted_fastqs/arbo_single_ms\n",
      "../human_depleted_fastqs/arbo_single_ms/4660_S3_L001/bwa_mem_mapping/4660_S3_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 260030 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 260030 reads\n",
      "../human_depleted_fastqs/arbo_single_ms/4798_S16_L001/bwa_mem_mapping/4798_S16_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 3042 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 3042 reads\n",
      "../human_depleted_fastqs/arbo_single_ms/4797_S17_L001/bwa_mem_mapping/4797_S17_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 2844 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 2844 reads\n",
      "../human_depleted_fastqs/arbo_single_ms/4656_S18_L001/bwa_mem_mapping/4656_S18_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 188792 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 188792 reads\n",
      "../human_depleted_fastqs/arbo_single_ms/4777_S13_L001/bwa_mem_mapping/4777_S13_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 32504 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 32504 reads\n",
      "../human_depleted_fastqs/arbo_single_ms/4716_S2_L001/bwa_mem_mapping/4716_S2_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 62358 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 62358 reads\n",
      "../human_depleted_fastqs/arbo_single_ms/4489_S15_L001/bwa_mem_mapping/4489_S15_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 4888 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 4888 reads\n",
      "../human_depleted_fastqs/arbo_single_ms/4688_S11_L001/bwa_mem_mapping/4688_S11_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 76098 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 76098 reads\n",
      "../human_depleted_fastqs/arbo_single_ms/4676_S7_L001/bwa_mem_mapping/4676_S7_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 106836 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 106836 reads\n",
      "../human_depleted_fastqs/arbo_single_ms/4653_S8_L001/bwa_mem_mapping/4653_S8_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 276902 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 276902 reads\n",
      "../human_depleted_fastqs/arbo_single_ms/4520_S4_L001/bwa_mem_mapping/4520_S4_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 224806 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 224806 reads\n",
      "../human_depleted_fastqs/arbo_single_ms/4673_S12_L001/bwa_mem_mapping/4673_S12_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 166074 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 166074 reads\n",
      "../human_depleted_fastqs/arbo_single_ms/4697_S6_L001/bwa_mem_mapping/4697_S6_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 82068 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 82068 reads\n",
      "../human_depleted_fastqs/arbo_single_ms/4787_S10_L001/bwa_mem_mapping/4787_S10_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 30948 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 30948 reads\n",
      "../human_depleted_fastqs/arbo_single_ms/4885_S1_L001/bwa_mem_mapping/4885_S1_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 29574 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 29574 reads\n",
      "../human_depleted_fastqs/arbo_single_ms/4707_S5_L001/bwa_mem_mapping/4707_S5_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 22264 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 22264 reads\n",
      "../human_depleted_fastqs/arbo_single_ms/4544_S19_L001/bwa_mem_mapping/4544_S19_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 6556 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 6556 reads\n",
      "../human_depleted_fastqs/arbo_single_ms/4861_S9_L001/bwa_mem_mapping/4861_S9_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 34694 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 34694 reads\n",
      "../human_depleted_fastqs/arbo_single_ms/4668_S14_L001/bwa_mem_mapping/4668_S14_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 79758 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 79758 reads\n",
      "- meta_is in ../human_depleted_fastqs/meta_is\n",
      "../human_depleted_fastqs/meta_is/4660_S3_L001/bwa_mem_mapping/4660_S3_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 18798 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 18798 reads\n",
      "../human_depleted_fastqs/meta_is/4489_S16_L001/bwa_mem_mapping/4489_S16_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 194 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 194 reads\n",
      "../human_depleted_fastqs/meta_is/4544_S20_L001/bwa_mem_mapping/4544_S20_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 4660 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 4660 reads\n",
      "../human_depleted_fastqs/meta_is/4777_S13_L001/bwa_mem_mapping/4777_S13_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 9310 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 9310 reads\n",
      "../human_depleted_fastqs/meta_is/4797_S18_L001/bwa_mem_mapping/4797_S18_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 184 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 184 reads\n",
      "../human_depleted_fastqs/meta_is/4716_S2_L001/bwa_mem_mapping/4716_S2_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 182 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 182 reads\n",
      "../human_depleted_fastqs/meta_is/4656_S19_L001/bwa_mem_mapping/4656_S19_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 104830 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 104830 reads\n",
      "../human_depleted_fastqs/meta_is/4688_S11_L001/bwa_mem_mapping/4688_S11_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 45066 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 45066 reads\n",
      "../human_depleted_fastqs/meta_is/4676_S7_L001/bwa_mem_mapping/4676_S7_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 14946 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 14946 reads\n",
      "../human_depleted_fastqs/meta_is/4653_S8_L001/bwa_mem_mapping/4653_S8_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 33266 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 33266 reads\n",
      "../human_depleted_fastqs/meta_is/4520_S4_L001/bwa_mem_mapping/4520_S4_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 6920 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 6920 reads\n",
      "../human_depleted_fastqs/meta_is/4673_S12_L001/bwa_mem_mapping/4673_S12_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 121650 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 121650 reads\n",
      "../human_depleted_fastqs/meta_is/4510_S15_L001/bwa_mem_mapping/4510_S15_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 1420 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 1420 reads\n",
      "../human_depleted_fastqs/meta_is/Controle-Negatif-water_S21_L001/bwa_mem_mapping/Controle-Negatif-water_S21_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 2 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 2 reads\n",
      "../human_depleted_fastqs/meta_is/4697_S6_L001/bwa_mem_mapping/4697_S6_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 97218 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 97218 reads\n",
      "../human_depleted_fastqs/meta_is/4787_S10_L001/bwa_mem_mapping/4787_S10_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 7510 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 7510 reads\n",
      "../human_depleted_fastqs/meta_is/4885_S1_L001/bwa_mem_mapping/4885_S1_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 456 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 456 reads\n",
      "../human_depleted_fastqs/meta_is/4707_S5_L001/bwa_mem_mapping/4707_S5_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 210 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 210 reads\n",
      "../human_depleted_fastqs/meta_is/4861_S9_L001/bwa_mem_mapping/4861_S9_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 438 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 438 reads\n",
      "../human_depleted_fastqs/meta_is/4798_S17_L001/bwa_mem_mapping/4798_S17_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 202 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 202 reads\n",
      "../human_depleted_fastqs/meta_is/4668_S14_L001/bwa_mem_mapping/4668_S14_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 14362 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 14362 reads\n",
      "- arbo_double_10plex_ms in ../human_depleted_fastqs/arbo_double_10plex_ms\n",
      "../human_depleted_fastqs/arbo_double_10plex_ms/4688-M10_S7_L001/bwa_mem_mapping/4688-M10_S7_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 9480 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 9480 reads\n",
      "../human_depleted_fastqs/arbo_double_10plex_ms/4777-M10_S9_L001/bwa_mem_mapping/4777-M10_S9_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 356 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 356 reads\n",
      "../human_depleted_fastqs/arbo_double_10plex_ms/4707-M10_S1_L001/bwa_mem_mapping/4707-M10_S1_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 885404 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 885404 reads\n",
      "../human_depleted_fastqs/arbo_double_10plex_ms/4697-M10_S2_L001/bwa_mem_mapping/4697-M10_S2_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 50146 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 50146 reads\n",
      "../human_depleted_fastqs/arbo_double_10plex_ms/4861-M10_S5_L001/bwa_mem_mapping/4861-M10_S5_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 496102 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 496102 reads\n",
      "../human_depleted_fastqs/arbo_double_10plex_ms/4673-M10_S8_L001/bwa_mem_mapping/4673-M10_S8_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 21532 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 21532 reads\n",
      "../human_depleted_fastqs/arbo_double_10plex_ms/4489-M10_S10_L001/bwa_mem_mapping/4489-M10_S10_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 21146 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 21146 reads\n",
      "../human_depleted_fastqs/arbo_double_10plex_ms/4787-M10_S6_L001/bwa_mem_mapping/4787-M10_S6_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 9204 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 9204 reads\n",
      "../human_depleted_fastqs/arbo_double_10plex_ms/4653-M10_S4_L001/bwa_mem_mapping/4653-M10_S4_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 3022984 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 3022984 reads\n",
      "../human_depleted_fastqs/arbo_double_10plex_ms/4676-M10_S3_L001/bwa_mem_mapping/4676-M10_S3_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 1241314 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 1241314 reads\n",
      "- twist_cov_LB_ms in ../human_depleted_fastqs/twist_cov_LB_ms\n",
      "../human_depleted_fastqs/twist_cov_LB_ms/4660_S3_L001/bwa_mem_mapping/4660_S3_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 2710716 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 2710716 reads\n",
      "../human_depleted_fastqs/twist_cov_LB_ms/4489_S16_L001/bwa_mem_mapping/4489_S16_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 11868 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 11868 reads\n",
      "../human_depleted_fastqs/twist_cov_LB_ms/4544_S20_L001/bwa_mem_mapping/4544_S20_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 32910 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 32910 reads\n",
      "../human_depleted_fastqs/twist_cov_LB_ms/4777_S13_L001/bwa_mem_mapping/4777_S13_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 15580 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 15580 reads\n",
      "../human_depleted_fastqs/twist_cov_LB_ms/4797_S18_L001/bwa_mem_mapping/4797_S18_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 496 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 496 reads\n",
      "../human_depleted_fastqs/twist_cov_LB_ms/4716_S2_L001/bwa_mem_mapping/4716_S2_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 115070 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 115070 reads\n",
      "../human_depleted_fastqs/twist_cov_LB_ms/4656_S19_L001/bwa_mem_mapping/4656_S19_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 778642 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 778642 reads\n",
      "../human_depleted_fastqs/twist_cov_LB_ms/4688_S11_L001/bwa_mem_mapping/4688_S11_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 119858 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 119858 reads\n",
      "../human_depleted_fastqs/twist_cov_LB_ms/4676_S7_L001/bwa_mem_mapping/4676_S7_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 609350 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 609350 reads\n",
      "../human_depleted_fastqs/twist_cov_LB_ms/4653_S8_L001/bwa_mem_mapping/4653_S8_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 1902070 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 1902070 reads\n",
      "../human_depleted_fastqs/twist_cov_LB_ms/4520_S4_L001/bwa_mem_mapping/4520_S4_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 1102598 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 1102598 reads\n",
      "../human_depleted_fastqs/twist_cov_LB_ms/4673_S12_L001/bwa_mem_mapping/4673_S12_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 506234 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 506234 reads\n",
      "../human_depleted_fastqs/twist_cov_LB_ms/4510_S15_L001/bwa_mem_mapping/4510_S15_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 281750 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 281750 reads\n",
      "../human_depleted_fastqs/twist_cov_LB_ms/4697_S6_L001/bwa_mem_mapping/4697_S6_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 101980 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 101980 reads\n",
      "../human_depleted_fastqs/twist_cov_LB_ms/4787_S10_L001/bwa_mem_mapping/4787_S10_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 16930 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 16930 reads\n",
      "../human_depleted_fastqs/twist_cov_LB_ms/4885_S1_L001/bwa_mem_mapping/4885_S1_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 328974 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 328974 reads\n",
      "../human_depleted_fastqs/twist_cov_LB_ms/4707_S5_L001/bwa_mem_mapping/4707_S5_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 61808 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 61808 reads\n",
      "../human_depleted_fastqs/twist_cov_LB_ms/4861_S9_L001/bwa_mem_mapping/4861_S9_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 291720 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 291720 reads\n",
      "../human_depleted_fastqs/twist_cov_LB_ms/4798_S17_L001/bwa_mem_mapping/4798_S17_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 454 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 454 reads\n",
      "../human_depleted_fastqs/twist_cov_LB_ms/4668_S14_L001/bwa_mem_mapping/4668_S14_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 14820 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 14820 reads\n",
      "- twist_cov_ms in ../human_depleted_fastqs/twist_cov_ms\n",
      "../human_depleted_fastqs/twist_cov_ms/4660_S3_L001/bwa_mem_mapping/4660_S3_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 398236 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 398236 reads\n",
      "../human_depleted_fastqs/twist_cov_ms/4489_S16_L001/bwa_mem_mapping/4489_S16_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 29388 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 29388 reads\n",
      "../human_depleted_fastqs/twist_cov_ms/4544_S20_L001/bwa_mem_mapping/4544_S20_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 41728 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 41728 reads\n",
      "../human_depleted_fastqs/twist_cov_ms/4777_S13_L001/bwa_mem_mapping/4777_S13_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 41726 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 41726 reads\n",
      "../human_depleted_fastqs/twist_cov_ms/4797_S18_L001/bwa_mem_mapping/4797_S18_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 9132 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 9132 reads\n",
      "../human_depleted_fastqs/twist_cov_ms/4716_S2_L001/bwa_mem_mapping/4716_S2_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 97060 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 97060 reads\n",
      "../human_depleted_fastqs/twist_cov_ms/4656_S19_L001/bwa_mem_mapping/4656_S19_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 693742 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 693742 reads\n",
      "../human_depleted_fastqs/twist_cov_ms/4688_S11_L001/bwa_mem_mapping/4688_S11_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 198668 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 198668 reads\n",
      "../human_depleted_fastqs/twist_cov_ms/4676_S7_L001/bwa_mem_mapping/4676_S7_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 211438 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 211438 reads\n",
      "../human_depleted_fastqs/twist_cov_ms/4653_S8_L001/bwa_mem_mapping/4653_S8_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 871772 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 871772 reads\n",
      "../human_depleted_fastqs/twist_cov_ms/4520_S4_L001/bwa_mem_mapping/4520_S4_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 333430 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 333430 reads\n",
      "../human_depleted_fastqs/twist_cov_ms/4673_S12_L001/bwa_mem_mapping/4673_S12_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 384778 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 384778 reads\n",
      "../human_depleted_fastqs/twist_cov_ms/4510_S15_L001/bwa_mem_mapping/4510_S15_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 189070 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 189070 reads\n",
      "../human_depleted_fastqs/twist_cov_ms/4697_S6_L001/bwa_mem_mapping/4697_S6_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 161988 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 161988 reads\n",
      "../human_depleted_fastqs/twist_cov_ms/4787_S10_L001/bwa_mem_mapping/4787_S10_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 22512 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 22512 reads\n",
      "../human_depleted_fastqs/twist_cov_ms/4885_S1_L001/bwa_mem_mapping/4885_S1_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 70322 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 70322 reads\n",
      "../human_depleted_fastqs/twist_cov_ms/4707_S5_L001/bwa_mem_mapping/4707_S5_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 40668 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 40668 reads\n",
      "../human_depleted_fastqs/twist_cov_ms/4861_S9_L001/bwa_mem_mapping/4861_S9_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 358520 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 358520 reads\n",
      "../human_depleted_fastqs/twist_cov_ms/4798_S17_L001/bwa_mem_mapping/4798_S17_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 9524 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 9524 reads\n",
      "../human_depleted_fastqs/twist_cov_ms/4668_S14_L001/bwa_mem_mapping/4668_S14_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 49668 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 49668 reads\n",
      "- illumina_resv_is in ../human_depleted_fastqs/illumina_resv_is\n",
      "../human_depleted_fastqs/illumina_resv_is/4660_S3_L001/bwa_mem_mapping/4660_S3_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 38846 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 38846 reads\n",
      "../human_depleted_fastqs/illumina_resv_is/4798_S16_L001/bwa_mem_mapping/4798_S16_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 798 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 798 reads\n",
      "../human_depleted_fastqs/illumina_resv_is/4797_S17_L001/bwa_mem_mapping/4797_S17_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 1024 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 1024 reads\n",
      "../human_depleted_fastqs/illumina_resv_is/4656_S18_L001/bwa_mem_mapping/4656_S18_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 9478 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 9478 reads\n",
      "../human_depleted_fastqs/illumina_resv_is/4777_S13_L001/bwa_mem_mapping/4777_S13_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 3224 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 3224 reads\n",
      "../human_depleted_fastqs/illumina_resv_is/4716_S2_L001/bwa_mem_mapping/4716_S2_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 8110 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 8110 reads\n",
      "../human_depleted_fastqs/illumina_resv_is/4489_S15_L001/bwa_mem_mapping/4489_S15_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 848 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 848 reads\n",
      "../human_depleted_fastqs/illumina_resv_is/4688_S11_L001/bwa_mem_mapping/4688_S11_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 95944 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 95944 reads\n",
      "../human_depleted_fastqs/illumina_resv_is/4676_S7_L001/bwa_mem_mapping/4676_S7_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 14202 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 14202 reads\n",
      "../human_depleted_fastqs/illumina_resv_is/4653_S8_L001/bwa_mem_mapping/4653_S8_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 36742 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 36742 reads\n",
      "../human_depleted_fastqs/illumina_resv_is/4520_S4_L001/bwa_mem_mapping/4520_S4_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 38874 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 38874 reads\n",
      "../human_depleted_fastqs/illumina_resv_is/4673_S12_L001/bwa_mem_mapping/4673_S12_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 12124 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 12124 reads\n",
      "../human_depleted_fastqs/illumina_resv_is/4697_S6_L001/bwa_mem_mapping/4697_S6_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 21278 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 21278 reads\n",
      "../human_depleted_fastqs/illumina_resv_is/4787_S10_L001/bwa_mem_mapping/4787_S10_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 1332 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 1332 reads\n",
      "../human_depleted_fastqs/illumina_resv_is/4885_S1_L001/bwa_mem_mapping/4885_S1_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 6300 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 6300 reads\n",
      "../human_depleted_fastqs/illumina_resv_is/4707_S5_L001/bwa_mem_mapping/4707_S5_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 2074 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 2074 reads\n",
      "../human_depleted_fastqs/illumina_resv_is/4544_S19_L001/bwa_mem_mapping/4544_S19_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 210234 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 210234 reads\n",
      "../human_depleted_fastqs/illumina_resv_is/4861_S9_L001/bwa_mem_mapping/4861_S9_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 3540 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 3540 reads\n",
      "../human_depleted_fastqs/illumina_resv_is/4668_S14_L001/bwa_mem_mapping/4668_S14_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 3474 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 3474 reads\n",
      "- twist_panviral_ns in ../human_depleted_fastqs/twist_panviral_ns\n",
      "../human_depleted_fastqs/twist_panviral_ns/4489_S15/bwa_mem_mapping/4489_S15.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 12004 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 12004 reads\n",
      "../human_depleted_fastqs/twist_panviral_ns/4656_S18/bwa_mem_mapping/4656_S18.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 145698 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 145698 reads\n",
      "../human_depleted_fastqs/twist_panviral_ns/4797_S17/bwa_mem_mapping/4797_S17.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 15950 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 15950 reads\n",
      "../human_depleted_fastqs/twist_panviral_ns/4520_S4/bwa_mem_mapping/4520_S4.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 111258 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 111258 reads\n",
      "../human_depleted_fastqs/twist_panviral_ns/4787_S10/bwa_mem_mapping/4787_S10.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 30364 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 30364 reads\n",
      "../human_depleted_fastqs/twist_panviral_ns/4777_S13/bwa_mem_mapping/4777_S13.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 60758 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 60758 reads\n",
      "../human_depleted_fastqs/twist_panviral_ns/4697_S6/bwa_mem_mapping/4697_S6.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 376572 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 376572 reads\n",
      "../human_depleted_fastqs/twist_panviral_ns/4707_S5/bwa_mem_mapping/4707_S5.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 10398 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 10398 reads\n",
      "../human_depleted_fastqs/twist_panviral_ns/4688_S11/bwa_mem_mapping/4688_S11.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 2405750 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 2405750 reads\n",
      "../human_depleted_fastqs/twist_panviral_ns/4861_S9/bwa_mem_mapping/4861_S9.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 27030 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 27030 reads\n",
      "../human_depleted_fastqs/twist_panviral_ns/4673_S12/bwa_mem_mapping/4673_S12.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 273008 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 273008 reads\n",
      "../human_depleted_fastqs/twist_panviral_ns/4885_S1/bwa_mem_mapping/4885_S1.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 12072 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 12072 reads\n",
      "../human_depleted_fastqs/twist_panviral_ns/4660_S3/bwa_mem_mapping/4660_S3.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 271758 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 271758 reads\n",
      "../human_depleted_fastqs/twist_panviral_ns/4544_S19/bwa_mem_mapping/4544_S19.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 3531732 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 3531732 reads\n",
      "../human_depleted_fastqs/twist_panviral_ns/4798_S16/bwa_mem_mapping/4798_S16.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 14048 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 14048 reads\n",
      "../human_depleted_fastqs/twist_panviral_ns/4668_S14/bwa_mem_mapping/4668_S14.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 162118 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 162118 reads\n",
      "../human_depleted_fastqs/twist_panviral_ns/4716_S2/bwa_mem_mapping/4716_S2.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 24108 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 24108 reads\n",
      "../human_depleted_fastqs/twist_panviral_ns/4676_S7/bwa_mem_mapping/4676_S7.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 139590 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 139590 reads\n",
      "../human_depleted_fastqs/twist_panviral_ns/4653_S8/bwa_mem_mapping/4653_S8.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 274934 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 274934 reads\n",
      "- illumina_resv_ns in ../human_depleted_fastqs/illumina_resv_ns\n",
      "../human_depleted_fastqs/illumina_resv_ns/4489_S15/bwa_mem_mapping/4489_S15.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 102266 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 102266 reads\n",
      "../human_depleted_fastqs/illumina_resv_ns/4656_S18/bwa_mem_mapping/4656_S18.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 802654 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 802654 reads\n",
      "../human_depleted_fastqs/illumina_resv_ns/4797_S17/bwa_mem_mapping/4797_S17.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 75506 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 75506 reads\n",
      "../human_depleted_fastqs/illumina_resv_ns/4520_S4/bwa_mem_mapping/4520_S4.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 4502330 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 4502330 reads\n",
      "../human_depleted_fastqs/illumina_resv_ns/4787_S10/bwa_mem_mapping/4787_S10.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 119986 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 119986 reads\n",
      "../human_depleted_fastqs/illumina_resv_ns/4777_S13/bwa_mem_mapping/4777_S13.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 283026 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 283026 reads\n",
      "../human_depleted_fastqs/illumina_resv_ns/4697_S6/bwa_mem_mapping/4697_S6.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 1820078 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 1820078 reads\n",
      "../human_depleted_fastqs/illumina_resv_ns/4707_S5/bwa_mem_mapping/4707_S5.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 238304 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 238304 reads\n",
      "../human_depleted_fastqs/illumina_resv_ns/4688_S11/bwa_mem_mapping/4688_S11.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 12871916 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 12871916 reads\n",
      "../human_depleted_fastqs/illumina_resv_ns/4861_S9/bwa_mem_mapping/4861_S9.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 278088 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 278088 reads\n",
      "../human_depleted_fastqs/illumina_resv_ns/4673_S12/bwa_mem_mapping/4673_S12.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 1070490 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 1070490 reads\n",
      "../human_depleted_fastqs/illumina_resv_ns/4885_S1/bwa_mem_mapping/4885_S1.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 572204 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 572204 reads\n",
      "../human_depleted_fastqs/illumina_resv_ns/4660_S3/bwa_mem_mapping/4660_S3.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 4293074 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 4293074 reads\n",
      "../human_depleted_fastqs/illumina_resv_ns/4544_S19/bwa_mem_mapping/4544_S19.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 26475222 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 26475222 reads\n",
      "../human_depleted_fastqs/illumina_resv_ns/4798_S16/bwa_mem_mapping/4798_S16.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 68294 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 68294 reads\n",
      "../human_depleted_fastqs/illumina_resv_ns/4668_S14/bwa_mem_mapping/4668_S14.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 337176 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 337176 reads\n",
      "../human_depleted_fastqs/illumina_resv_ns/4716_S2/bwa_mem_mapping/4716_S2.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 864854 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 864854 reads\n",
      "../human_depleted_fastqs/illumina_resv_ns/4676_S7/bwa_mem_mapping/4676_S7.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 1320964 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 1320964 reads\n",
      "../human_depleted_fastqs/illumina_resv_ns/4653_S8/bwa_mem_mapping/4653_S8.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 3483958 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 3483958 reads\n",
      "- arbo_double_ms in ../human_depleted_fastqs/arbo_double_ms\n",
      "../human_depleted_fastqs/arbo_double_ms/4660_S3_L001/bwa_mem_mapping/4660_S3_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 501784 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 501784 reads\n",
      "../human_depleted_fastqs/arbo_double_ms/4798_S16_L001/bwa_mem_mapping/4798_S16_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 1186 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 1186 reads\n",
      "../human_depleted_fastqs/arbo_double_ms/4797_S17_L001/bwa_mem_mapping/4797_S17_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 1508 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 1508 reads\n",
      "../human_depleted_fastqs/arbo_double_ms/4656_S18_L001/bwa_mem_mapping/4656_S18_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 14522 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 14522 reads\n",
      "../human_depleted_fastqs/arbo_double_ms/4777_S13_L001/bwa_mem_mapping/4777_S13_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 6204 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 6204 reads\n",
      "../human_depleted_fastqs/arbo_double_ms/4716_S2_L001/bwa_mem_mapping/4716_S2_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 171776 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 171776 reads\n",
      "../human_depleted_fastqs/arbo_double_ms/4489_S15_L001/bwa_mem_mapping/4489_S15_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 264236 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 264236 reads\n",
      "../human_depleted_fastqs/arbo_double_ms/4688_S11_L001/bwa_mem_mapping/4688_S11_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 5488 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 5488 reads\n",
      "../human_depleted_fastqs/arbo_double_ms/4676_S7_L001/bwa_mem_mapping/4676_S7_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 303020 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 303020 reads\n",
      "../human_depleted_fastqs/arbo_double_ms/4653_S8_L001/bwa_mem_mapping/4653_S8_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 868356 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 868356 reads\n",
      "../human_depleted_fastqs/arbo_double_ms/4520_S4_L001/bwa_mem_mapping/4520_S4_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 515276 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 515276 reads\n",
      "../human_depleted_fastqs/arbo_double_ms/4673_S12_L001/bwa_mem_mapping/4673_S12_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 55636 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 55636 reads\n",
      "../human_depleted_fastqs/arbo_double_ms/4697_S6_L001/bwa_mem_mapping/4697_S6_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 6462 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 6462 reads\n",
      "../human_depleted_fastqs/arbo_double_ms/4787_S10_L001/bwa_mem_mapping/4787_S10_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 6256 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 6256 reads\n",
      "../human_depleted_fastqs/arbo_double_ms/4885_S1_L001/bwa_mem_mapping/4885_S1_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 65392 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 65392 reads\n",
      "../human_depleted_fastqs/arbo_double_ms/4707_S5_L001/bwa_mem_mapping/4707_S5_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 166306 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 166306 reads\n",
      "../human_depleted_fastqs/arbo_double_ms/4544_S19_L001/bwa_mem_mapping/4544_S19_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 278 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 278 reads\n",
      "../human_depleted_fastqs/arbo_double_ms/4861_S9_L001/bwa_mem_mapping/4861_S9_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 1268838 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 1268838 reads\n",
      "../human_depleted_fastqs/arbo_double_ms/4668_S14_L001/bwa_mem_mapping/4668_S14_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 27436 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 27436 reads\n",
      "- illumina_resv_ms in ../human_depleted_fastqs/illumina_resv_ms\n",
      "../human_depleted_fastqs/illumina_resv_ms/4660_S3_L001/bwa_mem_mapping/4660_S3_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 157992 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 157992 reads\n",
      "../human_depleted_fastqs/illumina_resv_ms/4798_S16_L001/bwa_mem_mapping/4798_S16_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 3024 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 3024 reads\n",
      "../human_depleted_fastqs/illumina_resv_ms/4797_S17_L001/bwa_mem_mapping/4797_S17_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 4054 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 4054 reads\n",
      "../human_depleted_fastqs/illumina_resv_ms/4656_S18_L001/bwa_mem_mapping/4656_S18_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 38364 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 38364 reads\n",
      "../human_depleted_fastqs/illumina_resv_ms/4777_S13_L001/bwa_mem_mapping/4777_S13_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 13094 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 13094 reads\n",
      "../human_depleted_fastqs/illumina_resv_ms/4716_S2_L001/bwa_mem_mapping/4716_S2_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 30596 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 30596 reads\n",
      "../human_depleted_fastqs/illumina_resv_ms/4489_S15_L001/bwa_mem_mapping/4489_S15_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 3436 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 3436 reads\n",
      "../human_depleted_fastqs/illumina_resv_ms/4688_S11_L001/bwa_mem_mapping/4688_S11_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 383224 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 383224 reads\n",
      "../human_depleted_fastqs/illumina_resv_ms/4676_S7_L001/bwa_mem_mapping/4676_S7_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 50230 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 50230 reads\n",
      "../human_depleted_fastqs/illumina_resv_ms/4653_S8_L001/bwa_mem_mapping/4653_S8_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 143158 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 143158 reads\n",
      "../human_depleted_fastqs/illumina_resv_ms/4520_S4_L001/bwa_mem_mapping/4520_S4_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 126752 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 126752 reads\n",
      "../human_depleted_fastqs/illumina_resv_ms/4673_S12_L001/bwa_mem_mapping/4673_S12_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 45608 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 45608 reads\n",
      "../human_depleted_fastqs/illumina_resv_ms/4697_S6_L001/bwa_mem_mapping/4697_S6_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 80704 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 80704 reads\n",
      "../human_depleted_fastqs/illumina_resv_ms/4787_S10_L001/bwa_mem_mapping/4787_S10_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 6326 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 6326 reads\n",
      "../human_depleted_fastqs/illumina_resv_ms/4885_S1_L001/bwa_mem_mapping/4885_S1_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 24396 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 24396 reads\n",
      "../human_depleted_fastqs/illumina_resv_ms/4707_S5_L001/bwa_mem_mapping/4707_S5_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 7592 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 7592 reads\n",
      "../human_depleted_fastqs/illumina_resv_ms/4544_S19_L001/bwa_mem_mapping/4544_S19_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 819862 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 819862 reads\n",
      "../human_depleted_fastqs/illumina_resv_ms/4861_S9_L001/bwa_mem_mapping/4861_S9_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 12510 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 12510 reads\n",
      "../human_depleted_fastqs/illumina_resv_ms/4668_S14_L001/bwa_mem_mapping/4668_S14_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 14130 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 14130 reads\n",
      "- twist_cov_is in ../human_depleted_fastqs/twist_cov_is\n",
      "../human_depleted_fastqs/twist_cov_is/4660_S3_L001/bwa_mem_mapping/4660_S3_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 118424 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 118424 reads\n",
      "../human_depleted_fastqs/twist_cov_is/4489_S16_L001/bwa_mem_mapping/4489_S16_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 5056 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 5056 reads\n",
      "../human_depleted_fastqs/twist_cov_is/4544_S20_L001/bwa_mem_mapping/4544_S20_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 6348 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 6348 reads\n",
      "../human_depleted_fastqs/twist_cov_is/4777_S13_L001/bwa_mem_mapping/4777_S13_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 10346 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 10346 reads\n",
      "../human_depleted_fastqs/twist_cov_is/4797_S18_L001/bwa_mem_mapping/4797_S18_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 254 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 254 reads\n",
      "../human_depleted_fastqs/twist_cov_is/4716_S2_L001/bwa_mem_mapping/4716_S2_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 22574 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 22574 reads\n",
      "../human_depleted_fastqs/twist_cov_is/4656_S19_L001/bwa_mem_mapping/4656_S19_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 162834 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 162834 reads\n",
      "../human_depleted_fastqs/twist_cov_is/4688_S11_L001/bwa_mem_mapping/4688_S11_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 49028 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 49028 reads\n",
      "../human_depleted_fastqs/twist_cov_is/4676_S7_L001/bwa_mem_mapping/4676_S7_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 65858 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 65858 reads\n",
      "../human_depleted_fastqs/twist_cov_is/4653_S8_L001/bwa_mem_mapping/4653_S8_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 216078 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 216078 reads\n",
      "../human_depleted_fastqs/twist_cov_is/4520_S4_L001/bwa_mem_mapping/4520_S4_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 115498 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 115498 reads\n",
      "../human_depleted_fastqs/twist_cov_is/4673_S12_L001/bwa_mem_mapping/4673_S12_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 110322 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 110322 reads\n",
      "../human_depleted_fastqs/twist_cov_is/4510_S15_L001/bwa_mem_mapping/4510_S15_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 59926 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 59926 reads\n",
      "../human_depleted_fastqs/twist_cov_is/4697_S6_L001/bwa_mem_mapping/4697_S6_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 38628 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 38628 reads\n",
      "../human_depleted_fastqs/twist_cov_is/4787_S10_L001/bwa_mem_mapping/4787_S10_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 4782 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 4782 reads\n",
      "../human_depleted_fastqs/twist_cov_is/4885_S1_L001/bwa_mem_mapping/4885_S1_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 17702 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 17702 reads\n",
      "../human_depleted_fastqs/twist_cov_is/4707_S5_L001/bwa_mem_mapping/4707_S5_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 12494 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 12494 reads\n",
      "../human_depleted_fastqs/twist_cov_is/4861_S9_L001/bwa_mem_mapping/4861_S9_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 93502 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 93502 reads\n",
      "../human_depleted_fastqs/twist_cov_is/4798_S17_L001/bwa_mem_mapping/4798_S17_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 242 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 242 reads\n",
      "../human_depleted_fastqs/twist_cov_is/4668_S14_L001/bwa_mem_mapping/4668_S14_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 13264 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 13264 reads\n",
      "- twist_cov_10plex_ms in ../human_depleted_fastqs/twist_cov_10plex_ms\n",
      "../human_depleted_fastqs/twist_cov_10plex_ms/4688-M10_S7_L001/bwa_mem_mapping/4688-M10_S7_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 396100 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 396100 reads\n",
      "../human_depleted_fastqs/twist_cov_10plex_ms/4777-M10_S9_L001/bwa_mem_mapping/4777-M10_S9_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 20460 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 20460 reads\n",
      "../human_depleted_fastqs/twist_cov_10plex_ms/4707-M10_S1_L001/bwa_mem_mapping/4707-M10_S1_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 200478 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 200478 reads\n",
      "../human_depleted_fastqs/twist_cov_10plex_ms/4697-M10_S2_L001/bwa_mem_mapping/4697-M10_S2_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 927040 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 927040 reads\n",
      "../human_depleted_fastqs/twist_cov_10plex_ms/4861-M10_S5_L001/bwa_mem_mapping/4861-M10_S5_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 193088 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 193088 reads\n",
      "../human_depleted_fastqs/twist_cov_10plex_ms/4673-M10_S8_L001/bwa_mem_mapping/4673-M10_S8_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 561386 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 561386 reads\n",
      "../human_depleted_fastqs/twist_cov_10plex_ms/4489-M10_S10_L001/bwa_mem_mapping/4489-M10_S10_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 12704 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 12704 reads\n",
      "../human_depleted_fastqs/twist_cov_10plex_ms/4787-M10_S6_L001/bwa_mem_mapping/4787-M10_S6_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 28646 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 28646 reads\n",
      "../human_depleted_fastqs/twist_cov_10plex_ms/4653-M10_S4_L001/bwa_mem_mapping/4653-M10_S4_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 2721900 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 2721900 reads\n",
      "../human_depleted_fastqs/twist_cov_10plex_ms/4676-M10_S3_L001/bwa_mem_mapping/4676-M10_S3_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 738110 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 738110 reads\n",
      "- arbo_double_is in ../human_depleted_fastqs/arbo_double_is\n",
      "../human_depleted_fastqs/arbo_double_is/4660_S3_L001/bwa_mem_mapping/4660_S3_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 122250 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 122250 reads\n",
      "../human_depleted_fastqs/arbo_double_is/4798_S16_L001/bwa_mem_mapping/4798_S16_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 192 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 192 reads\n",
      "../human_depleted_fastqs/arbo_double_is/4797_S17_L001/bwa_mem_mapping/4797_S17_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 320 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 320 reads\n",
      "../human_depleted_fastqs/arbo_double_is/4656_S18_L001/bwa_mem_mapping/4656_S18_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 3846 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 3846 reads\n",
      "../human_depleted_fastqs/arbo_double_is/4777_S13_L001/bwa_mem_mapping/4777_S13_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 1572 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 1572 reads\n",
      "../human_depleted_fastqs/arbo_double_is/4716_S2_L001/bwa_mem_mapping/4716_S2_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 43212 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 43212 reads\n",
      "../human_depleted_fastqs/arbo_double_is/4489_S15_L001/bwa_mem_mapping/4489_S15_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 54940 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 54940 reads\n",
      "../human_depleted_fastqs/arbo_double_is/4688_S11_L001/bwa_mem_mapping/4688_S11_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 1240 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 1240 reads\n",
      "../human_depleted_fastqs/arbo_double_is/4676_S7_L001/bwa_mem_mapping/4676_S7_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 78028 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 78028 reads\n",
      "../human_depleted_fastqs/arbo_double_is/4653_S8_L001/bwa_mem_mapping/4653_S8_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 231444 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 231444 reads\n",
      "../human_depleted_fastqs/arbo_double_is/4520_S4_L001/bwa_mem_mapping/4520_S4_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 135256 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 135256 reads\n",
      "../human_depleted_fastqs/arbo_double_is/4673_S12_L001/bwa_mem_mapping/4673_S12_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 14742 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 14742 reads\n",
      "../human_depleted_fastqs/arbo_double_is/4697_S6_L001/bwa_mem_mapping/4697_S6_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 1498 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 1498 reads\n",
      "../human_depleted_fastqs/arbo_double_is/4787_S10_L001/bwa_mem_mapping/4787_S10_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 1520 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 1520 reads\n",
      "../human_depleted_fastqs/arbo_double_is/4885_S1_L001/bwa_mem_mapping/4885_S1_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 16214 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 16214 reads\n",
      "../human_depleted_fastqs/arbo_double_is/4707_S5_L001/bwa_mem_mapping/4707_S5_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 39974 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 39974 reads\n",
      "../human_depleted_fastqs/arbo_double_is/4544_S19_L001/bwa_mem_mapping/4544_S19_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 42 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 42 reads\n",
      "../human_depleted_fastqs/arbo_double_is/4861_S9_L001/bwa_mem_mapping/4861_S9_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 352204 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 352204 reads\n",
      "../human_depleted_fastqs/arbo_double_is/4668_S14_L001/bwa_mem_mapping/4668_S14_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 6376 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 6376 reads\n",
      "- meta_ns in ../human_depleted_fastqs/meta_ns\n",
      "../human_depleted_fastqs/meta_ns/4798_S17/bwa_mem_mapping/4798_S17.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 46846 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 46846 reads\n",
      "../human_depleted_fastqs/meta_ns/4656_S19/bwa_mem_mapping/4656_S19.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 9836598 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 9836598 reads\n",
      "../human_depleted_fastqs/meta_ns/4520_S4/bwa_mem_mapping/4520_S4.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 656548 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 656548 reads\n",
      "../human_depleted_fastqs/meta_ns/4797_S18/bwa_mem_mapping/4797_S18.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 29162 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 29162 reads\n",
      "../human_depleted_fastqs/meta_ns/4787_S10/bwa_mem_mapping/4787_S10.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 656526 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 656526 reads\n",
      "../human_depleted_fastqs/meta_ns/4777_S13/bwa_mem_mapping/4777_S13.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 810836 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 810836 reads\n",
      "../human_depleted_fastqs/meta_ns/4697_S6/bwa_mem_mapping/4697_S6.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 9029718 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 9029718 reads\n",
      "../human_depleted_fastqs/meta_ns/Cneg_Water_S21/bwa_mem_mapping/Cneg_Water_S21.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 214 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 214 reads\n",
      "../human_depleted_fastqs/meta_ns/4707_S5/bwa_mem_mapping/4707_S5.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 30182 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 30182 reads\n",
      "../human_depleted_fastqs/meta_ns/4688_S11/bwa_mem_mapping/4688_S11.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 4584006 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 4584006 reads\n",
      "../human_depleted_fastqs/meta_ns/4861_S9/bwa_mem_mapping/4861_S9.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 71954 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 71954 reads\n",
      "../human_depleted_fastqs/meta_ns/4673_S12/bwa_mem_mapping/4673_S12.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 11270608 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 11270608 reads\n",
      "../human_depleted_fastqs/meta_ns/4885_S1/bwa_mem_mapping/4885_S1.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 53716 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 53716 reads\n",
      "../human_depleted_fastqs/meta_ns/4660_S3/bwa_mem_mapping/4660_S3.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 1936126 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 1936126 reads\n",
      "../human_depleted_fastqs/meta_ns/4510_S15/bwa_mem_mapping/4510_S15.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 138002 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 138002 reads\n",
      "../human_depleted_fastqs/meta_ns/4489_S16/bwa_mem_mapping/4489_S16.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 41124 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 41124 reads\n",
      "../human_depleted_fastqs/meta_ns/4668_S14/bwa_mem_mapping/4668_S14.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 1521818 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 1521818 reads\n",
      "../human_depleted_fastqs/meta_ns/4716_S2/bwa_mem_mapping/4716_S2.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 33288 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 33288 reads\n",
      "../human_depleted_fastqs/meta_ns/4676_S7/bwa_mem_mapping/4676_S7.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 1545976 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 1545976 reads\n",
      "../human_depleted_fastqs/meta_ns/4544_S20/bwa_mem_mapping/4544_S20.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 507026 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 507026 reads\n",
      "../human_depleted_fastqs/meta_ns/4653_S8/bwa_mem_mapping/4653_S8.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 3348912 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 3348912 reads\n",
      "- twist_panviral_is in ../human_depleted_fastqs/twist_panviral_is\n",
      "../human_depleted_fastqs/twist_panviral_is/4660_S3_L001/bwa_mem_mapping/4660_S3_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 5354 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 5354 reads\n",
      "../human_depleted_fastqs/twist_panviral_is/4798_S16_L001/bwa_mem_mapping/4798_S16_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 298 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 298 reads\n",
      "../human_depleted_fastqs/twist_panviral_is/4797_S17_L001/bwa_mem_mapping/4797_S17_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 350 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 350 reads\n",
      "../human_depleted_fastqs/twist_panviral_is/4656_S18_L001/bwa_mem_mapping/4656_S18_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 3304 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 3304 reads\n",
      "../human_depleted_fastqs/twist_panviral_is/4777_S13_L001/bwa_mem_mapping/4777_S13_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 1164 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 1164 reads\n",
      "../human_depleted_fastqs/twist_panviral_is/4716_S2_L001/bwa_mem_mapping/4716_S2_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 408 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 408 reads\n",
      "../human_depleted_fastqs/twist_panviral_is/4489_S15_L001/bwa_mem_mapping/4489_S15_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 174 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 174 reads\n",
      "../human_depleted_fastqs/twist_panviral_is/4688_S11_L001/bwa_mem_mapping/4688_S11_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 34018 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 34018 reads\n",
      "../human_depleted_fastqs/twist_panviral_is/4676_S7_L001/bwa_mem_mapping/4676_S7_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 2628 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 2628 reads\n",
      "../human_depleted_fastqs/twist_panviral_is/4653_S8_L001/bwa_mem_mapping/4653_S8_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 5572 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 5572 reads\n",
      "../human_depleted_fastqs/twist_panviral_is/4520_S4_L001/bwa_mem_mapping/4520_S4_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 1966 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 1966 reads\n",
      "../human_depleted_fastqs/twist_panviral_is/4673_S12_L001/bwa_mem_mapping/4673_S12_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 5388 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 5388 reads\n",
      "../human_depleted_fastqs/twist_panviral_is/4697_S6_L001/bwa_mem_mapping/4697_S6_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 7802 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 7802 reads\n",
      "../human_depleted_fastqs/twist_panviral_is/4787_S10_L001/bwa_mem_mapping/4787_S10_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 642 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 642 reads\n",
      "../human_depleted_fastqs/twist_panviral_is/4885_S1_L001/bwa_mem_mapping/4885_S1_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 258 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 258 reads\n",
      "../human_depleted_fastqs/twist_panviral_is/4707_S5_L001/bwa_mem_mapping/4707_S5_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 224 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 224 reads\n",
      "../human_depleted_fastqs/twist_panviral_is/4544_S19_L001/bwa_mem_mapping/4544_S19_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 54096 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 54096 reads\n",
      "../human_depleted_fastqs/twist_panviral_is/4861_S9_L001/bwa_mem_mapping/4861_S9_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 696 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 696 reads\n",
      "../human_depleted_fastqs/twist_panviral_is/4668_S14_L001/bwa_mem_mapping/4668_S14_L001.sorted.bam\n",
      "HHH\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 2800 reads\n",
      "[M::bam2fq_mainloop] discarded 0 singletons\n",
      "[M::bam2fq_mainloop] processed 2800 reads\n"
     ]
    }
   ],
   "source": [
    "depletion_cmd = f\"\"\"\n",
    "bwa mem {GENOME_HUMAN_PATH} ./art/data_1.fq ../art/data_2.fq -t 8 > temp.sam \\\n",
    "samtools view -bS temp.sam > temp.bam \\\n",
    "samtools view -b -f 12 -F 256 temp.bam > temp_bothReadsUnmapped.bam \\\n",
    "bioconvert bam2fastq temp_bothReadsUnmapped.bam test.fastq --force \\\n",
    "\"\"\"\n",
    "\n",
    "for run_dir in RUN_DIRS:\n",
    "    \n",
    "    # Prepare mapper outdir\n",
    "    DEPLETION_DIR.mkdir(exist_ok=True)\n",
    "    out_dir = DEPLETION_DIR / run_dir.stem\n",
    "    print(f\"- {run_dir.stem} in {out_dir}\")\n",
    "    FASTQ_DIR = out_dir / \"depleted_fastqs\"\n",
    "    FASTQ_DIR.mkdir(exist_ok=True)\n",
    "    \n",
    "    sequana_cmd = f\"sequana_mapper --slurm-memory 30G --reference {GENOME_HUMAN_PATH} --input-directory {run_dir} --working-directory {out_dir} --mapper bwa\"\n",
    "    ! module load sequana/prod && {sequana_cmd}\n",
    "    ! module load sequana/prod && cd {out_dir} && sbatch -q biomicspole -p biomicspole mapper.sh\n",
    "    \n",
    "    for bam in out_dir.glob(\"*/bwa_mem_mapping/*.bam\"):\n",
    "        print(bam)\n",
    "        fastq = FASTQ_DIR / bam.with_suffix('.depleted.fastq').name\n",
    "        samtools_cmd = f\"samtools view -b -f 12 -F 256 {bam} > {bam.with_suffix('.depleted.bam')}\"\n",
    "        ! {samtools_cmd}\n",
    "        bioconvert_cmd = f\"module load bioconvert && bioconvert bam2fastq {bam.with_suffix('.depleted.bam')} {fastq} --force\"\n",
    "        ! {bioconvert_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Mapping Human and Sars-Cov"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "! cat {GENOME_SARS_PATH} {GENOME_HUMAN_PATH} > {GENOME_HUMAN_SARS_PATH}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "- illumina_resv_v2_ns in ../mapper_human_sarscov2/illumina_resv_v2_ns\n",
      "Sequana command: sequana_mapper --slurm-memory 30G --reference ../resources/human_sars.fa --input-directory ../data/illumina_resv_v2_ns --working-directory ../mapper_human_sarscov2/illumina_resv_v2_ns --mapper bowtie2\n",
      "INFO    [sequana.pipelines_common]:  Welcome to Sequana pipelines suite (sequana.readthedocs.io)\n",
      "INFO    [sequana.pipelines_common]:  Welcome to Sequana pipelines suite (sequana.readthedocs.io)\n",
      "INFO    [sequana.pipelines_common]:  Found 38 files matching your input  pattern (*fastq.gz)\n",
      "INFO    [sequana.pipelines_common]:  Found 38 files matching your input  pattern (*fastq.gz)\n",
      "Check the script in ../mapper_human_sarscov2/illumina_resv_v2_ns/mapper.sh as well as the configuration file in ../mapper_human_sarscov2/illumina_resv_v2_ns/config.yaml.\n",
      "\n",
      "Once ready, execute the script mapper.sh using \n",
      "\n",
      "\tcd ../mapper_human_sarscov2/illumina_resv_v2_ns; sbatch mapper.sh\n",
      "\n",
      "\n",
      "INFO    [sequana.pipelines_common]:  A completion if possible with sequana_completion --name mapper\n",
      "INFO    [sequana.pipelines_common]:  A completion if possible with sequana_completion --name mapper\n",
      "Submitted batch job 42814371\n",
      "- arbo_single_is in ../mapper_human_sarscov2/arbo_single_is\n",
      "Sequana command: sequana_mapper --slurm-memory 30G --reference ../resources/human_sars.fa --input-directory ../data/arbo_single_is --working-directory ../mapper_human_sarscov2/arbo_single_is --mapper bowtie2\n",
      "INFO    [sequana.pipelines_common]:  Welcome to Sequana pipelines suite (sequana.readthedocs.io)\n",
      "INFO    [sequana.pipelines_common]:  Welcome to Sequana pipelines suite (sequana.readthedocs.io)\n",
      "INFO    [sequana.pipelines_common]:  Found 38 files matching your input  pattern (*fastq.gz)\n",
      "INFO    [sequana.pipelines_common]:  Found 38 files matching your input  pattern (*fastq.gz)\n",
      "Check the script in ../mapper_human_sarscov2/arbo_single_is/mapper.sh as well as the configuration file in ../mapper_human_sarscov2/arbo_single_is/config.yaml.\n",
      "\n",
      "Once ready, execute the script mapper.sh using \n",
      "\n",
      "\tcd ../mapper_human_sarscov2/arbo_single_is; sbatch mapper.sh\n",
      "\n",
      "\n",
      "INFO    [sequana.pipelines_common]:  A completion if possible with sequana_completion --name mapper\n",
      "INFO    [sequana.pipelines_common]:  A completion if possible with sequana_completion --name mapper\n",
      "Submitted batch job 42814472\n",
      "- twist_cov_ns in ../mapper_human_sarscov2/twist_cov_ns\n",
      "Sequana command: sequana_mapper --slurm-memory 30G --reference ../resources/human_sars.fa --input-directory ../data/twist_cov_ns --working-directory ../mapper_human_sarscov2/twist_cov_ns --mapper bowtie2\n",
      "INFO    [sequana.pipelines_common]:  Welcome to Sequana pipelines suite (sequana.readthedocs.io)\n",
      "INFO    [sequana.pipelines_common]:  Welcome to Sequana pipelines suite (sequana.readthedocs.io)\n",
      "INFO    [sequana.pipelines_common]:  Found 40 files matching your input  pattern (*fastq.gz)\n",
      "INFO    [sequana.pipelines_common]:  Found 40 files matching your input  pattern (*fastq.gz)\n",
      "Check the script in ../mapper_human_sarscov2/twist_cov_ns/mapper.sh as well as the configuration file in ../mapper_human_sarscov2/twist_cov_ns/config.yaml.\n",
      "\n",
      "Once ready, execute the script mapper.sh using \n",
      "\n",
      "\tcd ../mapper_human_sarscov2/twist_cov_ns; sbatch mapper.sh\n",
      "\n",
      "\n",
      "INFO    [sequana.pipelines_common]:  A completion if possible with sequana_completion --name mapper\n",
      "INFO    [sequana.pipelines_common]:  A completion if possible with sequana_completion --name mapper\n",
      "Submitted batch job 42814609\n",
      "- arbo_single_ms in ../mapper_human_sarscov2/arbo_single_ms\n",
      "Sequana command: sequana_mapper --slurm-memory 30G --reference ../resources/human_sars.fa --input-directory ../data/arbo_single_ms --working-directory ../mapper_human_sarscov2/arbo_single_ms --mapper bowtie2\n",
      "INFO    [sequana.pipelines_common]:  Welcome to Sequana pipelines suite (sequana.readthedocs.io)\n",
      "INFO    [sequana.pipelines_common]:  Welcome to Sequana pipelines suite (sequana.readthedocs.io)\n",
      "INFO    [sequana.pipelines_common]:  Found 38 files matching your input  pattern (*fastq.gz)\n",
      "INFO    [sequana.pipelines_common]:  Found 38 files matching your input  pattern (*fastq.gz)\n",
      "Check the script in ../mapper_human_sarscov2/arbo_single_ms/mapper.sh as well as the configuration file in ../mapper_human_sarscov2/arbo_single_ms/config.yaml.\n",
      "\n",
      "Once ready, execute the script mapper.sh using \n",
      "\n",
      "\tcd ../mapper_human_sarscov2/arbo_single_ms; sbatch mapper.sh\n",
      "\n",
      "\n",
      "INFO    [sequana.pipelines_common]:  A completion if possible with sequana_completion --name mapper\n",
      "INFO    [sequana.pipelines_common]:  A completion if possible with sequana_completion --name mapper\n",
      "Submitted batch job 42814698\n",
      "- meta_is in ../mapper_human_sarscov2/meta_is\n",
      "Sequana command: sequana_mapper --slurm-memory 30G --reference ../resources/human_sars.fa --input-directory ../data/meta_is --working-directory ../mapper_human_sarscov2/meta_is --mapper bowtie2\n",
      "INFO    [sequana.pipelines_common]:  Welcome to Sequana pipelines suite (sequana.readthedocs.io)\n",
      "INFO    [sequana.pipelines_common]:  Welcome to Sequana pipelines suite (sequana.readthedocs.io)\n",
      "INFO    [sequana.pipelines_common]:  Found 42 files matching your input  pattern (*fastq.gz)\n",
      "INFO    [sequana.pipelines_common]:  Found 42 files matching your input  pattern (*fastq.gz)\n",
      "Check the script in ../mapper_human_sarscov2/meta_is/mapper.sh as well as the configuration file in ../mapper_human_sarscov2/meta_is/config.yaml.\n",
      "\n",
      "Once ready, execute the script mapper.sh using \n",
      "\n",
      "\tcd ../mapper_human_sarscov2/meta_is; sbatch mapper.sh\n",
      "\n",
      "\n",
      "INFO    [sequana.pipelines_common]:  A completion if possible with sequana_completion --name mapper\n",
      "INFO    [sequana.pipelines_common]:  A completion if possible with sequana_completion --name mapper\n",
      "Submitted batch job 42814814\n",
      "- arbo_double_10plex_ms in ../mapper_human_sarscov2/arbo_double_10plex_ms\n",
      "Sequana command: sequana_mapper --slurm-memory 30G --reference ../resources/human_sars.fa --input-directory ../data/arbo_double_10plex_ms --working-directory ../mapper_human_sarscov2/arbo_double_10plex_ms --mapper bowtie2\n",
      "INFO    [sequana.pipelines_common]:  Welcome to Sequana pipelines suite (sequana.readthedocs.io)\n",
      "INFO    [sequana.pipelines_common]:  Welcome to Sequana pipelines suite (sequana.readthedocs.io)\n",
      "INFO    [sequana.pipelines_common]:  Found 20 files matching your input  pattern (*fastq.gz)\n",
      "INFO    [sequana.pipelines_common]:  Found 20 files matching your input  pattern (*fastq.gz)\n",
      "Check the script in ../mapper_human_sarscov2/arbo_double_10plex_ms/mapper.sh as well as the configuration file in ../mapper_human_sarscov2/arbo_double_10plex_ms/config.yaml.\n",
      "\n",
      "Once ready, execute the script mapper.sh using \n",
      "\n",
      "\tcd ../mapper_human_sarscov2/arbo_double_10plex_ms; sbatch mapper.sh\n",
      "\n",
      "\n",
      "INFO    [sequana.pipelines_common]:  A completion if possible with sequana_completion --name mapper\n",
      "INFO    [sequana.pipelines_common]:  A completion if possible with sequana_completion --name mapper\n",
      "Submitted batch job 42814931\n",
      "- twist_cov_hg38_bwt2_depletion_ns in ../mapper_human_sarscov2/twist_cov_hg38_bwt2_depletion_ns\n",
      "Sequana command: sequana_mapper --slurm-memory 30G --reference ../resources/human_sars.fa --input-directory ../data/twist_cov_hg38_bwt2_depletion_ns --working-directory ../mapper_human_sarscov2/twist_cov_hg38_bwt2_depletion_ns --mapper bowtie2\n",
      "INFO    [sequana.pipelines_common]:  Welcome to Sequana pipelines suite (sequana.readthedocs.io)\n",
      "INFO    [sequana.pipelines_common]:  Welcome to Sequana pipelines suite (sequana.readthedocs.io)\n",
      "INFO    [sequana.pipelines_common]:  Found 40 files matching your input  pattern (*fastq.gz)\n",
      "INFO    [sequana.pipelines_common]:  Found 40 files matching your input  pattern (*fastq.gz)\n",
      "Check the script in ../mapper_human_sarscov2/twist_cov_hg38_bwt2_depletion_ns/mapper.sh as well as the configuration file in ../mapper_human_sarscov2/twist_cov_hg38_bwt2_depletion_ns/config.yaml.\n",
      "\n",
      "Once ready, execute the script mapper.sh using \n",
      "\n",
      "\tcd ../mapper_human_sarscov2/twist_cov_hg38_bwt2_depletion_ns; sbatch mapper.sh\n",
      "\n",
      "\n",
      "INFO    [sequana.pipelines_common]:  A completion if possible with sequana_completion --name mapper\n",
      "INFO    [sequana.pipelines_common]:  A completion if possible with sequana_completion --name mapper\n",
      "Submitted batch job 42815047\n",
      "- twist_cov_LB_ms in ../mapper_human_sarscov2/twist_cov_LB_ms\n",
      "Sequana command: sequana_mapper --slurm-memory 30G --reference ../resources/human_sars.fa --input-directory ../data/twist_cov_LB_ms --working-directory ../mapper_human_sarscov2/twist_cov_LB_ms --mapper bowtie2\n",
      "INFO    [sequana.pipelines_common]:  Welcome to Sequana pipelines suite (sequana.readthedocs.io)\n",
      "INFO    [sequana.pipelines_common]:  Welcome to Sequana pipelines suite (sequana.readthedocs.io)\n",
      "INFO    [sequana.pipelines_common]:  Found 40 files matching your input  pattern (*fastq.gz)\n",
      "INFO    [sequana.pipelines_common]:  Found 40 files matching your input  pattern (*fastq.gz)\n",
      "Check the script in ../mapper_human_sarscov2/twist_cov_LB_ms/mapper.sh as well as the configuration file in ../mapper_human_sarscov2/twist_cov_LB_ms/config.yaml.\n",
      "\n",
      "Once ready, execute the script mapper.sh using \n",
      "\n",
      "\tcd ../mapper_human_sarscov2/twist_cov_LB_ms; sbatch mapper.sh\n",
      "\n",
      "\n",
      "INFO    [sequana.pipelines_common]:  A completion if possible with sequana_completion --name mapper\n",
      "INFO    [sequana.pipelines_common]:  A completion if possible with sequana_completion --name mapper\n",
      "Submitted batch job 42815156\n",
      "- twist_cov_ms in ../mapper_human_sarscov2/twist_cov_ms\n",
      "Sequana command: sequana_mapper --slurm-memory 30G --reference ../resources/human_sars.fa --input-directory ../data/twist_cov_ms --working-directory ../mapper_human_sarscov2/twist_cov_ms --mapper bowtie2\n",
      "INFO    [sequana.pipelines_common]:  Welcome to Sequana pipelines suite (sequana.readthedocs.io)\n",
      "INFO    [sequana.pipelines_common]:  Welcome to Sequana pipelines suite (sequana.readthedocs.io)\n",
      "INFO    [sequana.pipelines_common]:  Found 40 files matching your input  pattern (*fastq.gz)\n",
      "INFO    [sequana.pipelines_common]:  Found 40 files matching your input  pattern (*fastq.gz)\n",
      "Check the script in ../mapper_human_sarscov2/twist_cov_ms/mapper.sh as well as the configuration file in ../mapper_human_sarscov2/twist_cov_ms/config.yaml.\n",
      "\n",
      "Once ready, execute the script mapper.sh using \n",
      "\n",
      "\tcd ../mapper_human_sarscov2/twist_cov_ms; sbatch mapper.sh\n",
      "\n",
      "\n",
      "INFO    [sequana.pipelines_common]:  A completion if possible with sequana_completion --name mapper\n",
      "INFO    [sequana.pipelines_common]:  A completion if possible with sequana_completion --name mapper\n",
      "Submitted batch job 42815290\n",
      "- illumina_resv_is in ../mapper_human_sarscov2/illumina_resv_is\n",
      "Sequana command: sequana_mapper --slurm-memory 30G --reference ../resources/human_sars.fa --input-directory ../data/illumina_resv_is --working-directory ../mapper_human_sarscov2/illumina_resv_is --mapper bowtie2\n",
      "INFO    [sequana.pipelines_common]:  Welcome to Sequana pipelines suite (sequana.readthedocs.io)\n",
      "INFO    [sequana.pipelines_common]:  Welcome to Sequana pipelines suite (sequana.readthedocs.io)\n",
      "INFO    [sequana.pipelines_common]:  Found 38 files matching your input  pattern (*fastq.gz)\n",
      "INFO    [sequana.pipelines_common]:  Found 38 files matching your input  pattern (*fastq.gz)\n",
      "Check the script in ../mapper_human_sarscov2/illumina_resv_is/mapper.sh as well as the configuration file in ../mapper_human_sarscov2/illumina_resv_is/config.yaml.\n",
      "\n",
      "Once ready, execute the script mapper.sh using \n",
      "\n",
      "\tcd ../mapper_human_sarscov2/illumina_resv_is; sbatch mapper.sh\n",
      "\n",
      "\n",
      "INFO    [sequana.pipelines_common]:  A completion if possible with sequana_completion --name mapper\n",
      "INFO    [sequana.pipelines_common]:  A completion if possible with sequana_completion --name mapper\n",
      "Submitted batch job 42815433\n",
      "- twist_panviral_ns in ../mapper_human_sarscov2/twist_panviral_ns\n",
      "Sequana command: sequana_mapper --slurm-memory 30G --reference ../resources/human_sars.fa --input-directory ../data/twist_panviral_ns --working-directory ../mapper_human_sarscov2/twist_panviral_ns --mapper bowtie2\n",
      "INFO    [sequana.pipelines_common]:  Welcome to Sequana pipelines suite (sequana.readthedocs.io)\n",
      "INFO    [sequana.pipelines_common]:  Welcome to Sequana pipelines suite (sequana.readthedocs.io)\n",
      "INFO    [sequana.pipelines_common]:  Found 38 files matching your input  pattern (*fastq.gz)\n",
      "INFO    [sequana.pipelines_common]:  Found 38 files matching your input  pattern (*fastq.gz)\n",
      "Check the script in ../mapper_human_sarscov2/twist_panviral_ns/mapper.sh as well as the configuration file in ../mapper_human_sarscov2/twist_panviral_ns/config.yaml.\n",
      "\n",
      "Once ready, execute the script mapper.sh using \n",
      "\n",
      "\tcd ../mapper_human_sarscov2/twist_panviral_ns; sbatch mapper.sh\n",
      "\n",
      "\n",
      "INFO    [sequana.pipelines_common]:  A completion if possible with sequana_completion --name mapper\n",
      "INFO    [sequana.pipelines_common]:  A completion if possible with sequana_completion --name mapper\n",
      "Submitted batch job 42815587\n",
      "- illumina_resv_ns in ../mapper_human_sarscov2/illumina_resv_ns\n",
      "Sequana command: sequana_mapper --slurm-memory 30G --reference ../resources/human_sars.fa --input-directory ../data/illumina_resv_ns --working-directory ../mapper_human_sarscov2/illumina_resv_ns --mapper bowtie2\n",
      "INFO    [sequana.pipelines_common]:  Welcome to Sequana pipelines suite (sequana.readthedocs.io)\n",
      "INFO    [sequana.pipelines_common]:  Welcome to Sequana pipelines suite (sequana.readthedocs.io)\n",
      "INFO    [sequana.pipelines_common]:  Found 38 files matching your input  pattern (*fastq.gz)\n",
      "INFO    [sequana.pipelines_common]:  Found 38 files matching your input  pattern (*fastq.gz)\n",
      "Check the script in ../mapper_human_sarscov2/illumina_resv_ns/mapper.sh as well as the configuration file in ../mapper_human_sarscov2/illumina_resv_ns/config.yaml.\n",
      "\n",
      "Once ready, execute the script mapper.sh using \n",
      "\n",
      "\tcd ../mapper_human_sarscov2/illumina_resv_ns; sbatch mapper.sh\n",
      "\n",
      "\n",
      "INFO    [sequana.pipelines_common]:  A completion if possible with sequana_completion --name mapper\n",
      "INFO    [sequana.pipelines_common]:  A completion if possible with sequana_completion --name mapper\n",
      "Submitted batch job 42815691\n",
      "- arbo_double_ms in ../mapper_human_sarscov2/arbo_double_ms\n",
      "Sequana command: sequana_mapper --slurm-memory 30G --reference ../resources/human_sars.fa --input-directory ../data/arbo_double_ms --working-directory ../mapper_human_sarscov2/arbo_double_ms --mapper bowtie2\n",
      "INFO    [sequana.pipelines_common]:  Welcome to Sequana pipelines suite (sequana.readthedocs.io)\n",
      "INFO    [sequana.pipelines_common]:  Welcome to Sequana pipelines suite (sequana.readthedocs.io)\n",
      "INFO    [sequana.pipelines_common]:  Found 38 files matching your input  pattern (*fastq.gz)\n",
      "INFO    [sequana.pipelines_common]:  Found 38 files matching your input  pattern (*fastq.gz)\n",
      "Check the script in ../mapper_human_sarscov2/arbo_double_ms/mapper.sh as well as the configuration file in ../mapper_human_sarscov2/arbo_double_ms/config.yaml.\n",
      "\n",
      "Once ready, execute the script mapper.sh using \n",
      "\n",
      "\tcd ../mapper_human_sarscov2/arbo_double_ms; sbatch mapper.sh\n",
      "\n",
      "\n",
      "INFO    [sequana.pipelines_common]:  A completion if possible with sequana_completion --name mapper\n",
      "INFO    [sequana.pipelines_common]:  A completion if possible with sequana_completion --name mapper\n",
      "Submitted batch job 42815842\n",
      "- illumina_resv_ms in ../mapper_human_sarscov2/illumina_resv_ms\n",
      "Sequana command: sequana_mapper --slurm-memory 30G --reference ../resources/human_sars.fa --input-directory ../data/illumina_resv_ms --working-directory ../mapper_human_sarscov2/illumina_resv_ms --mapper bowtie2\n",
      "INFO    [sequana.pipelines_common]:  Welcome to Sequana pipelines suite (sequana.readthedocs.io)\n",
      "INFO    [sequana.pipelines_common]:  Welcome to Sequana pipelines suite (sequana.readthedocs.io)\n",
      "INFO    [sequana.pipelines_common]:  Found 38 files matching your input  pattern (*fastq.gz)\n",
      "INFO    [sequana.pipelines_common]:  Found 38 files matching your input  pattern (*fastq.gz)\n",
      "Check the script in ../mapper_human_sarscov2/illumina_resv_ms/mapper.sh as well as the configuration file in ../mapper_human_sarscov2/illumina_resv_ms/config.yaml.\n",
      "\n",
      "Once ready, execute the script mapper.sh using \n",
      "\n",
      "\tcd ../mapper_human_sarscov2/illumina_resv_ms; sbatch mapper.sh\n",
      "\n",
      "\n",
      "INFO    [sequana.pipelines_common]:  A completion if possible with sequana_completion --name mapper\n",
      "INFO    [sequana.pipelines_common]:  A completion if possible with sequana_completion --name mapper\n",
      "Submitted batch job 42815933\n",
      "- twist_cov_is in ../mapper_human_sarscov2/twist_cov_is\n",
      "Sequana command: sequana_mapper --slurm-memory 30G --reference ../resources/human_sars.fa --input-directory ../data/twist_cov_is --working-directory ../mapper_human_sarscov2/twist_cov_is --mapper bowtie2\n",
      "INFO    [sequana.pipelines_common]:  Welcome to Sequana pipelines suite (sequana.readthedocs.io)\n",
      "INFO    [sequana.pipelines_common]:  Welcome to Sequana pipelines suite (sequana.readthedocs.io)\n",
      "INFO    [sequana.pipelines_common]:  Found 40 files matching your input  pattern (*fastq.gz)\n",
      "INFO    [sequana.pipelines_common]:  Found 40 files matching your input  pattern (*fastq.gz)\n",
      "Check the script in ../mapper_human_sarscov2/twist_cov_is/mapper.sh as well as the configuration file in ../mapper_human_sarscov2/twist_cov_is/config.yaml.\n",
      "\n",
      "Once ready, execute the script mapper.sh using \n",
      "\n",
      "\tcd ../mapper_human_sarscov2/twist_cov_is; sbatch mapper.sh\n",
      "\n",
      "\n",
      "INFO    [sequana.pipelines_common]:  A completion if possible with sequana_completion --name mapper\n",
      "INFO    [sequana.pipelines_common]:  A completion if possible with sequana_completion --name mapper\n",
      "Submitted batch job 42816041\n",
      "- twist_cov_10plex_ms in ../mapper_human_sarscov2/twist_cov_10plex_ms\n",
      "Sequana command: sequana_mapper --slurm-memory 30G --reference ../resources/human_sars.fa --input-directory ../data/twist_cov_10plex_ms --working-directory ../mapper_human_sarscov2/twist_cov_10plex_ms --mapper bowtie2\n",
      "INFO    [sequana.pipelines_common]:  Welcome to Sequana pipelines suite (sequana.readthedocs.io)\n",
      "INFO    [sequana.pipelines_common]:  Welcome to Sequana pipelines suite (sequana.readthedocs.io)\n",
      "INFO    [sequana.pipelines_common]:  Found 20 files matching your input  pattern (*fastq.gz)\n",
      "INFO    [sequana.pipelines_common]:  Found 20 files matching your input  pattern (*fastq.gz)\n",
      "Check the script in ../mapper_human_sarscov2/twist_cov_10plex_ms/mapper.sh as well as the configuration file in ../mapper_human_sarscov2/twist_cov_10plex_ms/config.yaml.\n",
      "\n",
      "Once ready, execute the script mapper.sh using \n",
      "\n",
      "\tcd ../mapper_human_sarscov2/twist_cov_10plex_ms; sbatch mapper.sh\n",
      "\n",
      "\n",
      "INFO    [sequana.pipelines_common]:  A completion if possible with sequana_completion --name mapper\n",
      "INFO    [sequana.pipelines_common]:  A completion if possible with sequana_completion --name mapper\n",
      "Submitted batch job 42816149\n",
      "- arbo_double_is in ../mapper_human_sarscov2/arbo_double_is\n",
      "Sequana command: sequana_mapper --slurm-memory 30G --reference ../resources/human_sars.fa --input-directory ../data/arbo_double_is --working-directory ../mapper_human_sarscov2/arbo_double_is --mapper bowtie2\n",
      "INFO    [sequana.pipelines_common]:  Welcome to Sequana pipelines suite (sequana.readthedocs.io)\n",
      "INFO    [sequana.pipelines_common]:  Welcome to Sequana pipelines suite (sequana.readthedocs.io)\n",
      "INFO    [sequana.pipelines_common]:  Found 38 files matching your input  pattern (*fastq.gz)\n",
      "INFO    [sequana.pipelines_common]:  Found 38 files matching your input  pattern (*fastq.gz)\n",
      "Check the script in ../mapper_human_sarscov2/arbo_double_is/mapper.sh as well as the configuration file in ../mapper_human_sarscov2/arbo_double_is/config.yaml.\n",
      "\n",
      "Once ready, execute the script mapper.sh using \n",
      "\n",
      "\tcd ../mapper_human_sarscov2/arbo_double_is; sbatch mapper.sh\n",
      "\n",
      "\n",
      "INFO    [sequana.pipelines_common]:  A completion if possible with sequana_completion --name mapper\n",
      "INFO    [sequana.pipelines_common]:  A completion if possible with sequana_completion --name mapper\n",
      "Submitted batch job 42816275\n",
      "- meta_ns in ../mapper_human_sarscov2/meta_ns\n",
      "Sequana command: sequana_mapper --slurm-memory 30G --reference ../resources/human_sars.fa --input-directory ../data/meta_ns --working-directory ../mapper_human_sarscov2/meta_ns --mapper bowtie2\n",
      "INFO    [sequana.pipelines_common]:  Welcome to Sequana pipelines suite (sequana.readthedocs.io)\n",
      "INFO    [sequana.pipelines_common]:  Welcome to Sequana pipelines suite (sequana.readthedocs.io)\n",
      "INFO    [sequana.pipelines_common]:  Found 42 files matching your input  pattern (*fastq.gz)\n",
      "INFO    [sequana.pipelines_common]:  Found 42 files matching your input  pattern (*fastq.gz)\n",
      "Check the script in ../mapper_human_sarscov2/meta_ns/mapper.sh as well as the configuration file in ../mapper_human_sarscov2/meta_ns/config.yaml.\n",
      "\n",
      "Once ready, execute the script mapper.sh using \n",
      "\n",
      "\tcd ../mapper_human_sarscov2/meta_ns; sbatch mapper.sh\n",
      "\n",
      "\n",
      "INFO    [sequana.pipelines_common]:  A completion if possible with sequana_completion --name mapper\n",
      "INFO    [sequana.pipelines_common]:  A completion if possible with sequana_completion --name mapper\n",
      "Submitted batch job 42816381\n",
      "- twist_panviral_is in ../mapper_human_sarscov2/twist_panviral_is\n",
      "Sequana command: sequana_mapper --slurm-memory 30G --reference ../resources/human_sars.fa --input-directory ../data/twist_panviral_is --working-directory ../mapper_human_sarscov2/twist_panviral_is --mapper bowtie2\n",
      "INFO    [sequana.pipelines_common]:  Welcome to Sequana pipelines suite (sequana.readthedocs.io)\n",
      "INFO    [sequana.pipelines_common]:  Welcome to Sequana pipelines suite (sequana.readthedocs.io)\n",
      "INFO    [sequana.pipelines_common]:  Found 38 files matching your input  pattern (*fastq.gz)\n",
      "INFO    [sequana.pipelines_common]:  Found 38 files matching your input  pattern (*fastq.gz)\n",
      "Check the script in ../mapper_human_sarscov2/twist_panviral_is/mapper.sh as well as the configuration file in ../mapper_human_sarscov2/twist_panviral_is/config.yaml.\n",
      "\n",
      "Once ready, execute the script mapper.sh using \n",
      "\n",
      "\tcd ../mapper_human_sarscov2/twist_panviral_is; sbatch mapper.sh\n",
      "\n",
      "\n",
      "INFO    [sequana.pipelines_common]:  A completion if possible with sequana_completion --name mapper\n",
      "INFO    [sequana.pipelines_common]:  A completion if possible with sequana_completion --name mapper\n",
      "Submitted batch job 42816506\n"
     ]
    }
   ],
   "source": [
    "for run_dir in RUN_DIRS:\n",
    "    \n",
    "    # Prepare mapper outdir\n",
    "    run_dir = Path(run_dir)\n",
    "    out_dir = MAPPER_HUMAN_SARS_DIR / run_dir.stem\n",
    "    print(f\"- {run_dir.stem} in {out_dir}\")\n",
    "    \n",
    "    # Check if mapping already done:\n",
    "    if out_dir.exists():\n",
    "        print(f\"The result folder {out_dir} exists already. Skipping...\")\n",
    "        continue\n",
    "        \n",
    "    # Prepare sequana command\n",
    "    sequana_cmd = f\"sequana_mapper --slurm-memory 30G --reference {GENOME_HUMAN_SARS_PATH} --input-directory {run_dir} --working-directory {out_dir} --mapper bowtie2\"\n",
    "    print(f\"Sequana command: {sequana_cmd}\")\n",
    "    \n",
    "    # Create sequana templates\n",
    "    ! module load sequana/prod && {sequana_cmd}\n",
    "    ! module load sequana/prod && cd {out_dir} && sbatch -q hubbioit -p hubbioit mapper.sh"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python [conda env:dev]",
   "language": "python",
   "name": "conda-env-dev-py"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.3"
  },
  "toc-autonumbering": false,
  "toc-showcode": false,
  "toc-showmarkdowntxt": false
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
